# 의존성 다운
- pip 및 pandas import

In [1]:
!pip install pandas openpyxl

In [2]:
import pandas as pd
# 파일다운
from google.colab import files

## 파일개수 불러온 기록
- 총raw: 312개
- 2021 기준 서울 산부인과 397개였으니 코로나 겪고 얼추 맞는 수일듯?
- 광고 등 변수로 밀려 크롤링된게 있어 각각 정제 후 merging

In [3]:
#엑셀 읽어오기
df1 = pd.read_excel('./산부인과1.xlsx')
df2 = pd.read_excel('./산부인과2.xlsx')
df3 = pd.read_excel('./산부인과3.xlsx')
df4 = pd.read_excel('./산부인과4.xlsx')
df5 = pd.read_excel('./산부인과5.xlsx')

FileNotFoundError: [Errno 2] No such file or directory: './산부인과1.xlsx'

In [ ]:
#2021기준 397개라는데 네이버맵 싹긁어온건 312개.
total = len(df1) + len(df2) + len(df3) + len(df4) + len(df5)
total

In [ ]:
df5

# df1 라벨정리


## 필요한 라벨 : 5,10,17,21,22,23,24,26 (총8개)
## 필요없는 라벨: 1\~4 6\~9, 11\~16, 18\~20,25 + 26이후의 행들(27~)

- 필요한 라벨
1. LABEL-5: 이름
2. ~~ LABE-6: 포인트? ~~
3. ~~ LABEL-8: 진료시작시간~~
4. LABEL-10: 개괄주소
5. LABEL-17: 예약가능여부
6. LABEL-21: 진료시작시각
7. LABEL-22: 방문자 리뷰개수
8. LABEL-23: 블로그 리뷰개수
9. LABEL-24: 산부인과 전문의 수
10.  LABEL-26: 산부인과의외 전문의 여부


- 필요없는라벨: 1, 2, 3,4,
- 7: 진료시간에 따른 폐쇄여부
- 9 : 현재위치에서 몇km
- 11: 상세주소 열기
- 12: 24와 중복(24는 전체기재)
- 13,14,15,16 -> 거리뷰관련 옵션 다떨구기
- 18,19,20 : 진료시간기본내용(무의미)
- 25: 문의 여부


In [ ]:
#엑셀 읽어오기
df1 = pd.read_excel('./산부인과1.xlsx')
df1

LABEL-1                                            LABEL-2  \
0       광고  https://pcmap.place.naver.com/hospital/list?qu...   
1       광고  https://pcmap.place.naver.com/hospital/list?qu...   
2      NaN                                                NaN   
3      NaN                                                NaN   
4      NaN                                                NaN   
..     ...                                                ...   
71     NaN                                                NaN   
72     NaN                                                NaN   
73     NaN                                                NaN   
74     NaN                                                NaN   
75     NaN                                                NaN   

                                              LABEL-3  LABEL-4  \
0   https://search.pstatic.net/common/?autoRotate=...  의154081   
1   https://search.pstatic.net/common/?autoRotate=...  의126702   
2   https://search.pstatic.net/common/?autoRotate=...      NaN   
3   https://search.pstatic.net/common/?autoRotate=...      NaN   
4   https://search.pstatic.net/common/?autoRotate=...      NaN   
..                                                ...      ...   
71                                                NaN      NaN   
72                                                NaN      NaN   
73                                                NaN      NaN   
74                                                NaN      NaN   
75                                                NaN      NaN   

             LABEL-5           LABEL-6 LABEL-7       LABEL-8       LABEL-9  \
0          김은주산부인과의원   여의사임신수술여성성형여성검진   진료 종료  10:00에 진료 시작   현재 위치에서15km   
1         나를위한산부인과의원  산부인과전문의 여의사 여성성형   진료 종료  10:00에 진료 시작  현재 위치에서9.0km   
2       애플산부인과의원 강남점               NaN     NaN           NaN  현재 위치에서8.7km   
3          에이린산부인과의원               NaN     NaN           NaN  현재 위치에서8.4km   
4        애플산부인과의원 잠실               NaN     NaN           NaN  현재 위치에서8.2km   
..               ...               ...     ...           ...           ...   
71            제이여성병원               NaN     NaN           NaN  현재 위치에서8.3km   
72   연세더블유산부인과의원 여의도               NaN     NaN           NaN   현재 위치에서12km   
73         엠제이산부인과의원               NaN     NaN           NaN   현재 위치에서17km   
74          닥터오즈여성의원               NaN     NaN           NaN  현재 위치에서8.6km   
75  유테라산부인과의원 서울대입구역               NaN     NaN           NaN   현재 위치에서13km   

        LABEL-10  ... LABEL-28 LABEL-29 LABEL-30 LABEL-31 LABEL-32 LABEL-33  \
0    서울 구로구 신도림동  ...      NaN      NaN      NaN      NaN      NaN      NaN   
1     서울 강남구 역삼동  ...      NaN      NaN      NaN      NaN      NaN      NaN   
2     서울 서초구 서초동  ...      NaN      NaN      NaN      NaN      NaN      NaN   
3     서울 서초구 서초동  ...      NaN      NaN      NaN      NaN      NaN      NaN   
4     서울 송파구 잠실동  ...      NaN      NaN      NaN      NaN      NaN      NaN   
..           ...  ...      ...      ...      ...      ...      ...      ...   
71   서울 서대문구 홍제동  ...      NaN      NaN      NaN      NaN      NaN      NaN   
72  서울 영등포구 여의도동  ...      NaN      NaN      NaN      NaN      NaN      NaN   
73    서울 금천구 독산동  ...      NaN      NaN      NaN      NaN      NaN      NaN   
74    서울 송파구 신천동  ...      NaN      NaN      NaN      NaN      NaN      NaN   
75    서울 관악구 봉천동  ...      NaN      NaN      NaN      NaN      NaN      NaN   

   LABEL-34  LABEL-35 LABEL-36 LABEL-37  
0       NaN       NaN      NaN      NaN  
1       NaN       NaN      NaN      NaN  
2       NaN       NaN      NaN      NaN  
3       NaN       NaN      NaN      NaN  
4       NaN       NaN      NaN      NaN  
..      ...       ...      ...      ...  
71      NaN       NaN      NaN      NaN  
72      NaN       NaN      NaN      NaN  
73      NaN       NaN      NaN      NaN  
74      NaN       NaN      NaN      NaN  
75      NaN       NaN      NaN      NaN  

[76 rows x 37 columns]

In [ ]:
df1 = df1.drop(columns  = ['LABEL-1','LABEL-2','LABEL-3','LABEL-4','LABEL-6','LABEL-8'])
df1 = df1.drop(columns = ['LABEL-7','LABEL-9','LABEL-11','LABEL-12','LABEL-13','LABEL-14','LABEL-15','LABEL-16','LABEL-18','LABEL-19','LABEL-20','LABEL-25'])
df1

LABEL-5      LABEL-10 LABEL-17      LABEL-21      LABEL-22  \
0          김은주산부인과의원   서울 구로구 신도림동      NaN           NaN           NaN   
1         나를위한산부인과의원    서울 강남구 역삼동       예약           NaN           NaN   
2       애플산부인과의원 강남점    서울 서초구 서초동       예약  10:00에 진료 시작  방문자 리뷰 1,305   
3          에이린산부인과의원    서울 서초구 서초동       예약  10:00에 진료 시작    방문자 리뷰 394   
4        애플산부인과의원 잠실    서울 송파구 잠실동       예약      07/23 휴무    방문자 리뷰 958   
..               ...           ...      ...           ...           ...   
71            제이여성병원   서울 서대문구 홍제동      NaN  08:30에 진료 시작  방문자 리뷰 1,690   
72   연세더블유산부인과의원 여의도  서울 영등포구 여의도동       예약  10:00에 진료 시작    방문자 리뷰 820   
73         엠제이산부인과의원    서울 금천구 독산동      NaN  09:00에 진료 시작  방문자 리뷰 1,615   
74          닥터오즈여성의원    서울 송파구 신천동       예약  09:30에 진료 시작    방문자 리뷰 403   
75  유테라산부인과의원 서울대입구역    서울 관악구 봉천동       예약  10:00에 진료 시작  방문자 리뷰 1,334   

        LABEL-23    LABEL-24      LABEL-26  LABEL-27  LABEL-28  LABEL-29  \
0            NaN         NaN           NaN       NaN       NaN       NaN   
1            NaN         NaN           NaN       NaN       NaN       NaN   
2     블로그 리뷰 650  산부인과전문의 3명           NaN       NaN       NaN       NaN   
3   블로그 리뷰 1,353  산부인과전문의 1명           NaN       NaN       NaN       NaN   
4     블로그 리뷰 824  산부인과전문의 2명           NaN       NaN       NaN       NaN   
..           ...         ...           ...       ...       ...       ...   
71    블로그 리뷰 656  산부인과전문의 4명  소아청소년과전문의 1명       NaN       NaN       NaN   
72    블로그 리뷰 249  산부인과전문의 1명           NaN       NaN       NaN       NaN   
73    블로그 리뷰 588  산부인과전문의 4명           NaN       NaN       NaN       NaN   
74    블로그 리뷰 791  산부인과전문의 1명           NaN       NaN       NaN       NaN   
75     블로그 리뷰 85  산부인과전문의 3명           NaN       NaN       NaN       NaN   

    LABEL-30  LABEL-31  LABEL-32  LABEL-33  LABEL-34  LABEL-35  LABEL-36  \
0        NaN       NaN       NaN       NaN       NaN       NaN       NaN   
1        NaN       NaN       NaN       NaN       NaN       NaN       NaN   
2        NaN       NaN       NaN       NaN       NaN       NaN       NaN   
3        NaN       NaN       NaN       NaN       NaN       NaN       NaN   
4        NaN       NaN       NaN       NaN       NaN       NaN       NaN   
..       ...       ...       ...       ...       ...       ...       ...   
71       NaN       NaN       NaN       NaN       NaN       NaN       NaN   
72       NaN       NaN       NaN       NaN       NaN       NaN       NaN   
73       NaN       NaN       NaN       NaN       NaN       NaN       NaN   
74       NaN       NaN       NaN       NaN       NaN       NaN       NaN   
75       NaN       NaN       NaN       NaN       NaN       NaN       NaN   

    LABEL-37  
0        NaN  
1        NaN  
2        NaN  
3        NaN  
4        NaN  
..       ...  
71       NaN  
72       NaN  
73       NaN  
74       NaN  
75       NaN  

[76 rows x 19 columns]

In [ ]:
# 공백인 컬럼들
df1 = df1.drop(columns  = ['LABEL-27','LABEL-28','LABEL-29','LABEL-30','LABEL-31','LABEL-32','LABEL-33','LABEL-34','LABEL-35','LABEL-36','LABEL-37'])
df1

LABEL-5      LABEL-10 LABEL-17      LABEL-21      LABEL-22  \
0          김은주산부인과의원   서울 구로구 신도림동      NaN           NaN           NaN   
1         나를위한산부인과의원    서울 강남구 역삼동       예약           NaN           NaN   
2       애플산부인과의원 강남점    서울 서초구 서초동       예약  10:00에 진료 시작  방문자 리뷰 1,305   
3          에이린산부인과의원    서울 서초구 서초동       예약  10:00에 진료 시작    방문자 리뷰 394   
4        애플산부인과의원 잠실    서울 송파구 잠실동       예약      07/23 휴무    방문자 리뷰 958   
..               ...           ...      ...           ...           ...   
71            제이여성병원   서울 서대문구 홍제동      NaN  08:30에 진료 시작  방문자 리뷰 1,690   
72   연세더블유산부인과의원 여의도  서울 영등포구 여의도동       예약  10:00에 진료 시작    방문자 리뷰 820   
73         엠제이산부인과의원    서울 금천구 독산동      NaN  09:00에 진료 시작  방문자 리뷰 1,615   
74          닥터오즈여성의원    서울 송파구 신천동       예약  09:30에 진료 시작    방문자 리뷰 403   
75  유테라산부인과의원 서울대입구역    서울 관악구 봉천동       예약  10:00에 진료 시작  방문자 리뷰 1,334   

        LABEL-23    LABEL-24      LABEL-26  
0            NaN         NaN           NaN  
1            NaN         NaN           NaN  
2     블로그 리뷰 650  산부인과전문의 3명           NaN  
3   블로그 리뷰 1,353  산부인과전문의 1명           NaN  
4     블로그 리뷰 824  산부인과전문의 2명           NaN  
..           ...         ...           ...  
71    블로그 리뷰 656  산부인과전문의 4명  소아청소년과전문의 1명  
72    블로그 리뷰 249  산부인과전문의 1명           NaN  
73    블로그 리뷰 588  산부인과전문의 4명           NaN  
74    블로그 리뷰 791  산부인과전문의 1명           NaN  
75     블로그 리뷰 85  산부인과전문의 3명           NaN  

[76 rows x 8 columns]

In [ ]:
# df1= 컬럼8개
df1.rename(columns = {'LABEL-5':'병원명'}, inplace = True)
df1.rename(columns = {'LABEL-10':'주소'}, inplace = True)
df1.rename(columns = {'LABEL-17':'예약가능여부'}, inplace = True)
df1.rename(columns = {'LABEL-21':'진료시작시각'}, inplace = True)
df1.rename(columns = {'LABEL-22':'방문자리뷰'}, inplace = True)
df1.rename(columns = {'LABEL-23':'블로그리뷰'}, inplace = True)
df1.rename(columns = {'LABEL-24':'산부인과전문의수'}, inplace = True)
df1.rename(columns = {'LABEL-26':'기타전문의여부'}, inplace = True)
df1

병원명            주소 예약가능여부        진료시작시각         방문자리뷰  \
0          김은주산부인과의원   서울 구로구 신도림동    NaN           NaN           NaN   
1         나를위한산부인과의원    서울 강남구 역삼동     예약           NaN           NaN   
2       애플산부인과의원 강남점    서울 서초구 서초동     예약  10:00에 진료 시작  방문자 리뷰 1,305   
3          에이린산부인과의원    서울 서초구 서초동     예약  10:00에 진료 시작    방문자 리뷰 394   
4        애플산부인과의원 잠실    서울 송파구 잠실동     예약      07/23 휴무    방문자 리뷰 958   
..               ...           ...    ...           ...           ...   
71            제이여성병원   서울 서대문구 홍제동    NaN  08:30에 진료 시작  방문자 리뷰 1,690   
72   연세더블유산부인과의원 여의도  서울 영등포구 여의도동     예약  10:00에 진료 시작    방문자 리뷰 820   
73         엠제이산부인과의원    서울 금천구 독산동    NaN  09:00에 진료 시작  방문자 리뷰 1,615   
74          닥터오즈여성의원    서울 송파구 신천동     예약  09:30에 진료 시작    방문자 리뷰 403   
75  유테라산부인과의원 서울대입구역    서울 관악구 봉천동     예약  10:00에 진료 시작  방문자 리뷰 1,334   

           블로그리뷰    산부인과전문의수       기타전문의여부  
0            NaN         NaN           NaN  
1            NaN         NaN           NaN  
2     블로그 리뷰 650  산부인과전문의 3명           NaN  
3   블로그 리뷰 1,353  산부인과전문의 1명           NaN  
4     블로그 리뷰 824  산부인과전문의 2명           NaN  
..           ...         ...           ...  
71    블로그 리뷰 656  산부인과전문의 4명  소아청소년과전문의 1명  
72    블로그 리뷰 249  산부인과전문의 1명           NaN  
73    블로그 리뷰 588  산부인과전문의 4명           NaN  
74    블로그 리뷰 791  산부인과전문의 1명           NaN  
75     블로그 리뷰 85  산부인과전문의 3명           NaN  

[76 rows x 8 columns]

In [ ]:
for i in range(len(df1)):
  print(df1.iloc[i])

병원명           김은주산부인과의원
주소          서울 구로구 신도림동
예약가능여부              NaN
진료시작시각              NaN
방문자리뷰               NaN
블로그리뷰               NaN
산부인과전문의수            NaN
기타전문의여부             NaN
Name: 0, dtype: object
병원명         나를위한산부인과의원
주소          서울 강남구 역삼동
예약가능여부              예약
진료시작시각             NaN
방문자리뷰              NaN
블로그리뷰              NaN
산부인과전문의수           NaN
기타전문의여부            NaN
Name: 1, dtype: object
병원명         애플산부인과의원 강남점
주소            서울 서초구 서초동
예약가능여부                예약
진료시작시각      10:00에 진료 시작
방문자리뷰       방문자 리뷰 1,305
블로그리뷰         블로그 리뷰 650
산부인과전문의수      산부인과전문의 3명
기타전문의여부              NaN
Name: 2, dtype: object
병원명            에이린산부인과의원
주소            서울 서초구 서초동
예약가능여부                예약
진료시작시각      10:00에 진료 시작
방문자리뷰         방문자 리뷰 394
블로그리뷰       블로그 리뷰 1,353
산부인과전문의수      산부인과전문의 1명
기타전문의여부              NaN
Name: 3, dtype: object
병원명         애플산부인과의원 잠실
주소           서울 송파구 잠실동
예약가능여부               예약
진료시작시각         07/23 휴무
방문자리뷰        방문자 리뷰 958
블로그리뷰       

In [ ]:
# 파일다운
from google.colab import files

# 변경된 데이터를 엑셀 파일로 저장합니다
df1.to_excel('cleaned_df1.xlsx', index=False)

# 저장한 파일을 다운로드할 수 있도록 합니다
files.download('cleaned_df1.xlsx')

<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>

# 새 섹션

# json 변환법
- 아래의 방법으로 되면 산부인과test.json 방식으로 다운
- 아스키가 깨지는 경우
``` df1.to_json('파일명.json', orient='records', lines=True, force_ascii=False)``` 와 같이 force_ascii = False로 UTF-8로 엔코딩되게 해야함

In [ ]:
# JSON 파일로 저장합니다
df1.to_json('산부인과test.json', orient='records', lines=True)
print("JSON으로 변환 완료_Ascii")
files.download('산부인과test.json')

JSON으로 변환 완료


<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>

In [ ]:
# json 변환 시 UTF-8로 줘야함
# 데이터프레임을 JSON 파일로 저장 (UTF-8 인코딩)
df1.to_json('파일명.json', orient='records', lines=True, force_ascii=False)

# 저장한 파일을 Colab에서 다운로드할 수 있도록 합니다 (필요 시)
from google.colab import files
files.download('파일명.json')

<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>

# df2 정리

- 필요한 컬럼: 5,10,30,31,32,33,34,35
- 필요없는 컬럼:1\~4,6\~9, 11~29, 36, 37,38

In [ ]:
#엑셀 읽어오기
df2 = pd.read_excel('./산부인과2.xlsx')
df2

LABEL-1                                            LABEL-2  \
0       광고  https://pcmap.place.naver.com/hospital/list?qu...   
1       광고  https://pcmap.place.naver.com/hospital/list?qu...   
2      NaN                                                NaN   
3      NaN                                                NaN   
4      NaN                                                NaN   
..     ...                                                ...   
71     NaN                                                NaN   
72     NaN                                                NaN   
73     NaN                                                NaN   
74     NaN                                                NaN   
75     NaN                                                NaN   

                                              LABEL-3  LABEL-4        LABEL-5  \
0   https://search.pstatic.net/common/?autoRotate=...  의129323        리에스여성의원   
1   https://search.pstatic.net/common/?autoRotate=...  의125559        워커힐여성의원   
2                                                 NaN      NaN     도곡함춘산부인과의원   
3   https://search.pstatic.net/common/?autoRotate=...      NaN        포미즈여성병원   
4   https://search.pstatic.net/common/?autoRotate=...      NaN         루트여성의원   
..                                                ...      ...            ...   
71  https://search.pstatic.net/common/?autoRotate=...      NaN       이사벨라여성의원   
72  https://search.pstatic.net/common/?autoRotate=...      NaN         우리성심의원   
73  https://search.pstatic.net/common/?autoRotate=...      NaN           미즈의원   
74  https://search.pstatic.net/common/?autoRotate=...      NaN        솜산부인과의원   
75  https://search.pstatic.net/common/?autoRotate=...      NaN  헤스티아여성의원 신사본점   

             LABEL-6 LABEL-7  LABEL-8       LABEL-9    LABEL-10  ... LABEL-29  \
0   서울대의과대학 졸업 정창원원장   진료 종료      NaN  현재 위치에서6.7km  서울 강남구 논현동  ...      NaN   
1   여의사, 소음순수술, 여성성형   진료 종료      NaN  현재 위치에서5.3km  서울 광진구 구의동  ...      NaN   
2                NaN     NaN      NaN   현재 위치에서10km  서울 강남구 도곡동  ...    진료 종료   
3                NaN     NaN      NaN   현재 위치에서15km   서울 양천구 목동  ...    진료 종료   
4                NaN     NaN      NaN  현재 위치에서4.3km  서울 광진구 군자동  ...    진료 종료   
..               ...     ...      ...           ...         ...  ...      ...   
71               NaN     NaN      NaN  현재 위치에서9.6km  서울 강남구 대치동  ...    진료 종료   
72               NaN     NaN      NaN   현재 위치에서17km  서울 금천구 독산동  ...    진료 종료   
73               NaN     NaN      NaN   현재 위치에서11km  서울 은평구 불광동  ...    진료 종료   
74               NaN     NaN      NaN   현재 위치에서12km  서울 관악구 남현동  ...    진료 종료   
75               NaN     NaN      NaN  현재 위치에서6.7km  서울 강남구 신사동  ...    진료 종료   

        LABEL-30      LABEL-31      LABEL-32    LABEL-33       LABEL-34  \
0   10:00에 진료 시작           NaN           NaN  산부인과전문의 1명            NaN   
1   10:00에 진료 시작           NaN           NaN  산부인과전문의 1명  마취통증의학과전문의 1명   
2   09:00에 진료 시작    방문자 리뷰 377    블로그 리뷰 121         NaN            NaN   
3   09:00에 진료 시작  방문자 리뷰 1,974  블로그 리뷰 2,862  산부인과전문의 6명  내과전문의 1명 외 1명   
4   09:00에 진료 시작     방문자 리뷰 44      블로그 리뷰 9  산부인과전문의 3명            NaN   
..           ...           ...           ...         ...            ...   
71  09:30에 진료 시작    방문자 리뷰 136    블로그 리뷰 129  산부인과전문의 1명    가정의학과전문의 1명   
72  08:00에 진료 시작    방문자 리뷰 397    블로그 리뷰 234  산부인과전문의 1명       내과전문의 3명   
73  10:30에 진료 시작    방문자 리뷰 421    블로그 리뷰 859         NaN            NaN   
74  10:00에 진료 시작    방문자 리뷰 530  블로그 리뷰 1,284  산부인과전문의 1명            NaN   
75  10:00에 진료 시작    방문자 리뷰 538    블로그 리뷰 640  산부인과전문의 6명            NaN   

    LABEL-35  LABEL-36  LABEL-37  LABEL-38  
0        NaN       NaN       NaN       NaN  
1        NaN       NaN       NaN       NaN  
2        NaN       NaN       NaN       NaN  
3        NaN       NaN       NaN       NaN  
4         예약       NaN       NaN       NaN  
..       ...       ...       ...       ...  
71        예약       NaN       NaN       NaN  
72       NaN   

In [ ]:
df2 = df2.drop(columns = ['LABEL-11','LABEL-12','LABEL-13','LABEL-14','LABEL-15','LABEL-16','LABEL-17','LABEL-18','LABEL-19','LABEL-20'])
df2 = df2.drop(columns = ['LABEL-21','LABEL-22','LABEL-23','LABEL-24','LABEL-25','LABEL-26','LABEL-27','LABEL-28','LABEL-29'])
df2 = df2.drop(columns = ['LABEL-36','LABEL-37', 'LABEL-38'])
df2

LABEL-1                                            LABEL-2  \
0       광고  https://pcmap.place.naver.com/hospital/list?qu...   
1       광고  https://pcmap.place.naver.com/hospital/list?qu...   
2      NaN                                                NaN   
3      NaN                                                NaN   
4      NaN                                                NaN   
..     ...                                                ...   
71     NaN                                                NaN   
72     NaN                                                NaN   
73     NaN                                                NaN   
74     NaN                                                NaN   
75     NaN                                                NaN   

                                              LABEL-3  LABEL-4        LABEL-5  \
0   https://search.pstatic.net/common/?autoRotate=...  의129323        리에스여성의원   
1   https://search.pstatic.net/common/?autoRotate=...  의125559        워커힐여성의원   
2                                                 NaN      NaN     도곡함춘산부인과의원   
3   https://search.pstatic.net/common/?autoRotate=...      NaN        포미즈여성병원   
4   https://search.pstatic.net/common/?autoRotate=...      NaN         루트여성의원   
..                                                ...      ...            ...   
71  https://search.pstatic.net/common/?autoRotate=...      NaN       이사벨라여성의원   
72  https://search.pstatic.net/common/?autoRotate=...      NaN         우리성심의원   
73  https://search.pstatic.net/common/?autoRotate=...      NaN           미즈의원   
74  https://search.pstatic.net/common/?autoRotate=...      NaN        솜산부인과의원   
75  https://search.pstatic.net/common/?autoRotate=...      NaN  헤스티아여성의원 신사본점   

             LABEL-6 LABEL-7  LABEL-8       LABEL-9    LABEL-10      LABEL-30  \
0   서울대의과대학 졸업 정창원원장   진료 종료      NaN  현재 위치에서6.7km  서울 강남구 논현동  10:00에 진료 시작   
1   여의사, 소음순수술, 여성성형   진료 종료      NaN  현재 위치에서5.3km  서울 광진구 구의동  10:00에 진료 시작   
2                NaN     NaN      NaN   현재 위치에서10km  서울 강남구 도곡동  09:00에 진료 시작   
3                NaN     NaN      NaN   현재 위치에서15km   서울 양천구 목동  09:00에 진료 시작   
4                NaN     NaN      NaN  현재 위치에서4.3km  서울 광진구 군자동  09:00에 진료 시작   
..               ...     ...      ...           ...         ...           ...   
71               NaN     NaN      NaN  현재 위치에서9.6km  서울 강남구 대치동  09:30에 진료 시작   
72               NaN     NaN      NaN   현재 위치에서17km  서울 금천구 독산동  08:00에 진료 시작   
73               NaN     NaN      NaN   현재 위치에서11km  서울 은평구 불광동  10:30에 진료 시작   
74               NaN     NaN      NaN   현재 위치에서12km  서울 관악구 남현동  10:00에 진료 시작   
75               NaN     NaN      NaN  현재 위치에서6.7km  서울 강남구 신사동  10:00에 진료 시작   

        LABEL-31      LABEL-32    LABEL-33       LABEL-34 LABEL-35  
0            NaN           NaN  산부인과전문의 1명            NaN      NaN  
1            NaN           NaN  산부인과전문의 1명  마취통증의학과전문의 1명      NaN  
2     방문자 리뷰 377    블로그 리뷰 121         NaN            NaN      NaN  
3   방문자 리뷰 1,974  블로그 리뷰 2,862  산부인과전문의 6명  내과전문의 1명 외 1명      NaN  
4      방문자 리뷰 44      블로그 리뷰 9  산부인과전문의 3명            NaN       예약  
..           ...           ...         ...            ...      ...  
71    방문자 리뷰 136    블로그 리뷰 129  산부인과전문의 1명    가정의학과전문의 1명       예약  
72    방문자 리뷰 397    블로그 리뷰 234  산부인과전문의 1명       내과전문의 3명      NaN  
73    방문자 리뷰 421    블로그 리뷰 859         NaN            NaN       예약  
74    방문자 리뷰 530  블로그 리뷰 1,284  산부인과전문의 1명            NaN       예약  
75    방문자 리뷰 538    블로그 리뷰 640  산부인과전문의 6명            NaN      NaN  

[76 rows x 16 columns]

In [ ]:
df2 = df2.drop(columns  = ['LABEL-1','LABEL-2','LABEL-3','LABEL-4','LABEL-6','LABEL-7','LABEL-8','LABEL-9'])
df2

LABEL-5    LABEL-10      LABEL-30      LABEL-31      LABEL-32  \
0         리에스여성의원  서울 강남구 논현동  10:00에 진료 시작           NaN           NaN   
1         워커힐여성의원  서울 광진구 구의동  10:00에 진료 시작           NaN           NaN   
2      도곡함춘산부인과의원  서울 강남구 도곡동  09:00에 진료 시작    방문자 리뷰 377    블로그 리뷰 121   
3         포미즈여성병원   서울 양천구 목동  09:00에 진료 시작  방문자 리뷰 1,974  블로그 리뷰 2,862   
4          루트여성의원  서울 광진구 군자동  09:00에 진료 시작     방문자 리뷰 44      블로그 리뷰 9   
..            ...         ...           ...           ...           ...   
71       이사벨라여성의원  서울 강남구 대치동  09:30에 진료 시작    방문자 리뷰 136    블로그 리뷰 129   
72         우리성심의원  서울 금천구 독산동  08:00에 진료 시작    방문자 리뷰 397    블로그 리뷰 234   
73           미즈의원  서울 은평구 불광동  10:30에 진료 시작    방문자 리뷰 421    블로그 리뷰 859   
74        솜산부인과의원  서울 관악구 남현동  10:00에 진료 시작    방문자 리뷰 530  블로그 리뷰 1,284   
75  헤스티아여성의원 신사본점  서울 강남구 신사동  10:00에 진료 시작    방문자 리뷰 538    블로그 리뷰 640   

      LABEL-33       LABEL-34 LABEL-35  
0   산부인과전문의 1명            NaN      NaN  
1   산부인과전문의 1명  마취통증의학과전문의 1명      NaN  
2          NaN            NaN      NaN  
3   산부인과전문의 6명  내과전문의 1명 외 1명      NaN  
4   산부인과전문의 3명            NaN       예약  
..         ...            ...      ...  
71  산부인과전문의 1명    가정의학과전문의 1명       예약  
72  산부인과전문의 1명       내과전문의 3명      NaN  
73         NaN            NaN       예약  
74  산부인과전문의 1명            NaN       예약  
75  산부인과전문의 6명            NaN      NaN  

[76 rows x 8 columns]

In [ ]:
# df1= 컬럼8개 ,df2 = 컬럼8개
df2.rename(columns = {'LABEL-5':'병원명'}, inplace = True)
df2.rename(columns = {'LABEL-10':'주소'}, inplace = True)
df2.rename(columns = {'LABEL-35':'예약가능여부'}, inplace = True)
df2.rename(columns = {'LABEL-30':'진료시작시각'}, inplace = True)
df2.rename(columns = {'LABEL-31':'방문자리뷰'}, inplace = True)
df2.rename(columns = {'LABEL-32':'블로그리뷰'}, inplace = True)
df2.rename(columns = {'LABEL-33':'산부인과전문의수'}, inplace = True)
df2.rename(columns = {'LABEL-34':'기타전문의여부'}, inplace = True)
df2

병원명          주소        진료시작시각         방문자리뷰         블로그리뷰  \
0         리에스여성의원  서울 강남구 논현동  10:00에 진료 시작           NaN           NaN   
1         워커힐여성의원  서울 광진구 구의동  10:00에 진료 시작           NaN           NaN   
2      도곡함춘산부인과의원  서울 강남구 도곡동  09:00에 진료 시작    방문자 리뷰 377    블로그 리뷰 121   
3         포미즈여성병원   서울 양천구 목동  09:00에 진료 시작  방문자 리뷰 1,974  블로그 리뷰 2,862   
4          루트여성의원  서울 광진구 군자동  09:00에 진료 시작     방문자 리뷰 44      블로그 리뷰 9   
..            ...         ...           ...           ...           ...   
71       이사벨라여성의원  서울 강남구 대치동  09:30에 진료 시작    방문자 리뷰 136    블로그 리뷰 129   
72         우리성심의원  서울 금천구 독산동  08:00에 진료 시작    방문자 리뷰 397    블로그 리뷰 234   
73           미즈의원  서울 은평구 불광동  10:30에 진료 시작    방문자 리뷰 421    블로그 리뷰 859   
74        솜산부인과의원  서울 관악구 남현동  10:00에 진료 시작    방문자 리뷰 530  블로그 리뷰 1,284   
75  헤스티아여성의원 신사본점  서울 강남구 신사동  10:00에 진료 시작    방문자 리뷰 538    블로그 리뷰 640   

      산부인과전문의수        기타전문의여부 예약가능여부  
0   산부인과전문의 1명            NaN    NaN  
1   산부인과전문의 1명  마취통증의학과전문의 1명    NaN  
2          NaN            NaN    NaN  
3   산부인과전문의 6명  내과전문의 1명 외 1명    NaN  
4   산부인과전문의 3명            NaN     예약  
..         ...            ...    ...  
71  산부인과전문의 1명    가정의학과전문의 1명     예약  
72  산부인과전문의 1명       내과전문의 3명    NaN  
73         NaN            NaN     예약  
74  산부인과전문의 1명            NaN     예약  
75  산부인과전문의 6명            NaN    NaN  

[76 rows x 8 columns]

In [ ]:
df1

병원명            주소 예약가능여부        진료시작시각         방문자리뷰  \
0          김은주산부인과의원   서울 구로구 신도림동    NaN           NaN           NaN   
1         나를위한산부인과의원    서울 강남구 역삼동     예약           NaN           NaN   
2       애플산부인과의원 강남점    서울 서초구 서초동     예약  10:00에 진료 시작  방문자 리뷰 1,305   
3          에이린산부인과의원    서울 서초구 서초동     예약  10:00에 진료 시작    방문자 리뷰 394   
4        애플산부인과의원 잠실    서울 송파구 잠실동     예약      07/23 휴무    방문자 리뷰 958   
..               ...           ...    ...           ...           ...   
71            제이여성병원   서울 서대문구 홍제동    NaN  08:30에 진료 시작  방문자 리뷰 1,690   
72   연세더블유산부인과의원 여의도  서울 영등포구 여의도동     예약  10:00에 진료 시작    방문자 리뷰 820   
73         엠제이산부인과의원    서울 금천구 독산동    NaN  09:00에 진료 시작  방문자 리뷰 1,615   
74          닥터오즈여성의원    서울 송파구 신천동     예약  09:30에 진료 시작    방문자 리뷰 403   
75  유테라산부인과의원 서울대입구역    서울 관악구 봉천동     예약  10:00에 진료 시작  방문자 리뷰 1,334   

           블로그리뷰    산부인과전문의수       기타전문의여부  
0            NaN         NaN           NaN  
1            NaN         NaN           NaN  
2     블로그 리뷰 650  산부인과전문의 3명           NaN  
3   블로그 리뷰 1,353  산부인과전문의 1명           NaN  
4     블로그 리뷰 824  산부인과전문의 2명           NaN  
..           ...         ...           ...  
71    블로그 리뷰 656  산부인과전문의 4명  소아청소년과전문의 1명  
72    블로그 리뷰 249  산부인과전문의 1명           NaN  
73    블로그 리뷰 588  산부인과전문의 4명           NaN  
74    블로그 리뷰 791  산부인과전문의 1명           NaN  
75     블로그 리뷰 85  산부인과전문의 3명           NaN  

[76 rows x 8 columns]

In [ ]:
df2

병원명          주소        진료시작시각         방문자리뷰         블로그리뷰  \
0         리에스여성의원  서울 강남구 논현동  10:00에 진료 시작           NaN           NaN   
1         워커힐여성의원  서울 광진구 구의동  10:00에 진료 시작           NaN           NaN   
2      도곡함춘산부인과의원  서울 강남구 도곡동  09:00에 진료 시작    방문자 리뷰 377    블로그 리뷰 121   
3         포미즈여성병원   서울 양천구 목동  09:00에 진료 시작  방문자 리뷰 1,974  블로그 리뷰 2,862   
4          루트여성의원  서울 광진구 군자동  09:00에 진료 시작     방문자 리뷰 44      블로그 리뷰 9   
..            ...         ...           ...           ...           ...   
71       이사벨라여성의원  서울 강남구 대치동  09:30에 진료 시작    방문자 리뷰 136    블로그 리뷰 129   
72         우리성심의원  서울 금천구 독산동  08:00에 진료 시작    방문자 리뷰 397    블로그 리뷰 234   
73           미즈의원  서울 은평구 불광동  10:30에 진료 시작    방문자 리뷰 421    블로그 리뷰 859   
74        솜산부인과의원  서울 관악구 남현동  10:00에 진료 시작    방문자 리뷰 530  블로그 리뷰 1,284   
75  헤스티아여성의원 신사본점  서울 강남구 신사동  10:00에 진료 시작    방문자 리뷰 538    블로그 리뷰 640   

      산부인과전문의수        기타전문의여부 예약가능여부  
0   산부인과전문의 1명            NaN    NaN  
1   산부인과전문의 1명  마취통증의학과전문의 1명    NaN  
2          NaN            NaN    NaN  
3   산부인과전문의 6명  내과전문의 1명 외 1명    NaN  
4   산부인과전문의 3명            NaN     예약  
..         ...            ...    ...  
71  산부인과전문의 1명    가정의학과전문의 1명     예약  
72  산부인과전문의 1명       내과전문의 3명    NaN  
73         NaN            NaN     예약  
74  산부인과전문의 1명            NaN     예약  
75  산부인과전문의 6명            NaN    NaN  

[76 rows x 8 columns]

In [ ]:
#컬럼순서 맞춰주기
cols = ['병원명', '주소', '예약가능여부', '진료시작시각', '방문자리뷰', '블로그리뷰', '산부인과전문의수', '기타전문의여부']
df2 = df2[cols]
df2

병원명          주소 예약가능여부        진료시작시각         방문자리뷰  \
0         리에스여성의원  서울 강남구 논현동    NaN  10:00에 진료 시작           NaN   
1         워커힐여성의원  서울 광진구 구의동    NaN  10:00에 진료 시작           NaN   
2      도곡함춘산부인과의원  서울 강남구 도곡동    NaN  09:00에 진료 시작    방문자 리뷰 377   
3         포미즈여성병원   서울 양천구 목동    NaN  09:00에 진료 시작  방문자 리뷰 1,974   
4          루트여성의원  서울 광진구 군자동     예약  09:00에 진료 시작     방문자 리뷰 44   
..            ...         ...    ...           ...           ...   
71       이사벨라여성의원  서울 강남구 대치동     예약  09:30에 진료 시작    방문자 리뷰 136   
72         우리성심의원  서울 금천구 독산동    NaN  08:00에 진료 시작    방문자 리뷰 397   
73           미즈의원  서울 은평구 불광동     예약  10:30에 진료 시작    방문자 리뷰 421   
74        솜산부인과의원  서울 관악구 남현동     예약  10:00에 진료 시작    방문자 리뷰 530   
75  헤스티아여성의원 신사본점  서울 강남구 신사동    NaN  10:00에 진료 시작    방문자 리뷰 538   

           블로그리뷰    산부인과전문의수        기타전문의여부  
0            NaN  산부인과전문의 1명            NaN  
1            NaN  산부인과전문의 1명  마취통증의학과전문의 1명  
2     블로그 리뷰 121         NaN            NaN  
3   블로그 리뷰 2,862  산부인과전문의 6명  내과전문의 1명 외 1명  
4       블로그 리뷰 9  산부인과전문의 3명            NaN  
..           ...         ...            ...  
71    블로그 리뷰 129  산부인과전문의 1명    가정의학과전문의 1명  
72    블로그 리뷰 234  산부인과전문의 1명       내과전문의 3명  
73    블로그 리뷰 859         NaN            NaN  
74  블로그 리뷰 1,284  산부인과전문의 1명            NaN  
75    블로그 리뷰 640  산부인과전문의 6명            NaN  

[76 rows x 8 columns]

In [ ]:
df1

병원명            주소 예약가능여부        진료시작시각         방문자리뷰  \
0          김은주산부인과의원   서울 구로구 신도림동    NaN           NaN           NaN   
1         나를위한산부인과의원    서울 강남구 역삼동     예약           NaN           NaN   
2       애플산부인과의원 강남점    서울 서초구 서초동     예약  10:00에 진료 시작  방문자 리뷰 1,305   
3          에이린산부인과의원    서울 서초구 서초동     예약  10:00에 진료 시작    방문자 리뷰 394   
4        애플산부인과의원 잠실    서울 송파구 잠실동     예약      07/23 휴무    방문자 리뷰 958   
..               ...           ...    ...           ...           ...   
71            제이여성병원   서울 서대문구 홍제동    NaN  08:30에 진료 시작  방문자 리뷰 1,690   
72   연세더블유산부인과의원 여의도  서울 영등포구 여의도동     예약  10:00에 진료 시작    방문자 리뷰 820   
73         엠제이산부인과의원    서울 금천구 독산동    NaN  09:00에 진료 시작  방문자 리뷰 1,615   
74          닥터오즈여성의원    서울 송파구 신천동     예약  09:30에 진료 시작    방문자 리뷰 403   
75  유테라산부인과의원 서울대입구역    서울 관악구 봉천동     예약  10:00에 진료 시작  방문자 리뷰 1,334   

           블로그리뷰    산부인과전문의수       기타전문의여부  
0            NaN         NaN           NaN  
1            NaN         NaN           NaN  
2     블로그 리뷰 650  산부인과전문의 3명           NaN  
3   블로그 리뷰 1,353  산부인과전문의 1명           NaN  
4     블로그 리뷰 824  산부인과전문의 2명           NaN  
..           ...         ...           ...  
71    블로그 리뷰 656  산부인과전문의 4명  소아청소년과전문의 1명  
72    블로그 리뷰 249  산부인과전문의 1명           NaN  
73    블로그 리뷰 588  산부인과전문의 4명           NaN  
74    블로그 리뷰 791  산부인과전문의 1명           NaN  
75     블로그 리뷰 85  산부인과전문의 3명           NaN  

[76 rows x 8 columns]

## df1+ df2 merging

In [ ]:
# 두 데이터프레임을 병합합니다 (기본적으로 아래쪽으로 병합됨)
merged_df12 = pd.concat([df1, df2], ignore_index=True)
merged_df12

병원명           주소 예약가능여부        진료시작시각         방문자리뷰  \
0        김은주산부인과의원  서울 구로구 신도림동    NaN           NaN           NaN   
1       나를위한산부인과의원   서울 강남구 역삼동     예약           NaN           NaN   
2     애플산부인과의원 강남점   서울 서초구 서초동     예약  10:00에 진료 시작  방문자 리뷰 1,305   
3        에이린산부인과의원   서울 서초구 서초동     예약  10:00에 진료 시작    방문자 리뷰 394   
4      애플산부인과의원 잠실   서울 송파구 잠실동     예약      07/23 휴무    방문자 리뷰 958   
..             ...          ...    ...           ...           ...   
147       이사벨라여성의원   서울 강남구 대치동     예약  09:30에 진료 시작    방문자 리뷰 136   
148         우리성심의원   서울 금천구 독산동    NaN  08:00에 진료 시작    방문자 리뷰 397   
149           미즈의원   서울 은평구 불광동     예약  10:30에 진료 시작    방문자 리뷰 421   
150        솜산부인과의원   서울 관악구 남현동     예약  10:00에 진료 시작    방문자 리뷰 530   
151  헤스티아여성의원 신사본점   서울 강남구 신사동    NaN  10:00에 진료 시작    방문자 리뷰 538   

            블로그리뷰    산부인과전문의수      기타전문의여부  
0             NaN         NaN          NaN  
1             NaN         NaN          NaN  
2      블로그 리뷰 650  산부인과전문의 3명          NaN  
3    블로그 리뷰 1,353  산부인과전문의 1명          NaN  
4      블로그 리뷰 824  산부인과전문의 2명          NaN  
..            ...         ...          ...  
147    블로그 리뷰 129  산부인과전문의 1명  가정의학과전문의 1명  
148    블로그 리뷰 234  산부인과전문의 1명     내과전문의 3명  
149    블로그 리뷰 859         NaN          NaN  
150  블로그 리뷰 1,284  산부인과전문의 1명          NaN  
151    블로그 리뷰 640  산부인과전문의 6명          NaN  

[152 rows x 8 columns]

In [ ]:
# 파일다운
from google.colab import files

# 변경된 데이터를 엑셀 파일로 저장합니다
merged_df12.to_excel('merged_df12.xlsx', index=False)

# 저장한 파일을 다운로드할 수 있도록 합니다
files.download('merged_df12.xlsx')

<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>

# df3 정리

- 필요컬럼: 3, 6, 8, ,10,11,12,17, 18
- 필요없는 컬럼: 1,2,4,5,7,9,13~16,19,20

In [ ]:
#엑셀 읽어오기
df3 = pd.read_excel('./산부인과3.xlsx')
df3

LABEL-1  LABEL-2       LABEL-3  \
0   https://search.pstatic.net/common/?autoRotate=...      5.0       선산부인과의원   
1   https://search.pstatic.net/common/?autoRotate=...      5.0    밝은미래산부인과의원   
2   https://search.pstatic.net/common/?autoRotate=...     11.0    수서나을산부인과의원   
3   https://search.pstatic.net/common/?autoRotate=...      5.0     한순심산부인과의원   
4   https://search.pstatic.net/common/?autoRotate=...      5.0      그대의봄여성의원   
..                                                ...      ...           ...   
65                                                NaN      5.0       봄산부인과의원   
66                                                NaN      5.0  대학로메리힐산부인과의원   
67                                                NaN      5.0   연앤네이쳐산부인과의원   
68                                                NaN      4.0        로즈여성의원   
69                                                NaN      6.0  리즈산부인과의원 홍대점   

   LABEL-4 LABEL-5       LABEL-6       LABEL-7      LABEL-8  LABEL-9  \
0     산부인과   진료 종료  09:00에 진료 시작   현재 위치에서14km   서울 관악구 신림동  상세주소 열기   
1     산부인과   진료 종료  09:00에 진료 시작   현재 위치에서19km   서울 양천구 신정동  상세주소 열기   
2     산부인과   진료 종료  09:00에 진료 시작   현재 위치에서11km   서울 강남구 수서동  상세주소 열기   
3     산부인과   진료 종료  09:30에 진료 시작   현재 위치에서18km   서울 강서구 화곡동  상세주소 열기   
4     산부인과   진료 종료  09:00에 진료 시작  현재 위치에서4.3km    서울 광진구 능동  상세주소 열기   
..     ...     ...           ...           ...          ...      ...   
65    산부인과   진료 종료  09:00에 진료 시작  현재 위치에서8.5km  서울 서대문구 홍제동  상세주소 열기   
66    산부인과   진료 종료  10:00에 진료 시작  현재 위치에서3.2km   서울 종로구 동숭동  상세주소 열기   
67    산부인과   진료 종료  10:00에 진료 시작  현재 위치에서6.4km   서울 강남구 청담동  상세주소 열기   
68    산부인과   진료 종료  09:30에 진료 시작  현재 위치에서7.5km   서울 강남구 논현동  상세주소 열기   
69    산부인과   진료 종료  10:00에 진료 시작   현재 위치에서10km   서울 마포구 동교동  상세주소 열기   

        LABEL-10    LABEL-11    LABEL-12 LABEL-13 LABEL-14 LABEL-15 LABEL-16  \
0     방문자 리뷰 280  블로그 리뷰 102  산부인과전문의 1명       출발       도착      거리뷰      거리뷰   
1   방문자 리뷰 1,689  블로그 리뷰 209  산부인과전문의 4명       출발       도착      거리뷰      거리뷰   
2     방문자 리뷰 363   블로그 리뷰 53  산부인과전문의 3명       출발       도착      거리뷰      거리뷰   
3     방문자 리뷰 404  블로그 리뷰 587  산부인과전문의 1명       출발       도착      거리뷰      거리뷰   
4     방문자 리뷰 841  블로그 리뷰 263  산부인과전문의 2명       출발       도착      거리뷰      거리뷰   
..           ...         ...         ...      ...      ...      ...      ...   
65    방문자 리뷰 501    블로그 리뷰 9  산부인과전문의 1명       출발       도착      거리뷰      거리뷰   
66    방문자 리뷰 733         NaN  산부인과전문의 1명       출발       도착      거리뷰      거리뷰   
67    방문자 리뷰 144  블로그 리뷰 265         NaN       출발       도착      거리뷰      거리뷰   
68     방문자 리뷰 67  블로그 리뷰 600  산부인과전문의 2명       출발       도착      거리뷰      거리뷰   
69    방문자 리뷰 396   블로그 리뷰 17  산부인과전문의 1명       출발       도착      거리뷰      거리뷰   

   LABEL-17 LABEL-18 LABEL-19 LABEL-20  
0       NaN      NaN      NaN      NaN  
1       NaN      NaN      NaN      NaN  
2       NaN      NaN      NaN      NaN  
3       NaN      NaN      NaN      NaN  
4       NaN      NaN      NaN      NaN  
..      ...      ...      ...      ...  
65      NaN      NaN      NaN      NaN  
66      NaN      NaN      NaN      NaN  
67       예약      NaN      NaN      NaN  
68      NaN      NaN      NaN      NaN  
69       문의      NaN      NaN      NaN  

[70 rows x 20 columns]

In [ ]:
df3 = df3.drop(columns = ['LABEL-1','LABEL-2','LABEL-4','LABEL-5','LABEL-7','LABEL-9','LABEL-13','LABEL-14','LABEL-15','LABEL-16'])
df3 = df3.drop(columns = ['LABEL-19','LABEL-20'])
df3

LABEL-3       LABEL-6      LABEL-8      LABEL-10    LABEL-11  \
0        선산부인과의원  09:00에 진료 시작   서울 관악구 신림동    방문자 리뷰 280  블로그 리뷰 102   
1     밝은미래산부인과의원  09:00에 진료 시작   서울 양천구 신정동  방문자 리뷰 1,689  블로그 리뷰 209   
2     수서나을산부인과의원  09:00에 진료 시작   서울 강남구 수서동    방문자 리뷰 363   블로그 리뷰 53   
3      한순심산부인과의원  09:30에 진료 시작   서울 강서구 화곡동    방문자 리뷰 404  블로그 리뷰 587   
4       그대의봄여성의원  09:00에 진료 시작    서울 광진구 능동    방문자 리뷰 841  블로그 리뷰 263   
..           ...           ...          ...           ...         ...   
65       봄산부인과의원  09:00에 진료 시작  서울 서대문구 홍제동    방문자 리뷰 501    블로그 리뷰 9   
66  대학로메리힐산부인과의원  10:00에 진료 시작   서울 종로구 동숭동    방문자 리뷰 733         NaN   
67   연앤네이쳐산부인과의원  10:00에 진료 시작   서울 강남구 청담동    방문자 리뷰 144  블로그 리뷰 265   
68        로즈여성의원  09:30에 진료 시작   서울 강남구 논현동     방문자 리뷰 67  블로그 리뷰 600   
69  리즈산부인과의원 홍대점  10:00에 진료 시작   서울 마포구 동교동    방문자 리뷰 396   블로그 리뷰 17   

      LABEL-12 LABEL-17 LABEL-18  
0   산부인과전문의 1명      NaN      NaN  
1   산부인과전문의 4명      NaN      NaN  
2   산부인과전문의 3명      NaN      NaN  
3   산부인과전문의 1명      NaN      NaN  
4   산부인과전문의 2명      NaN      NaN  
..         ...      ...      ...  
65  산부인과전문의 1명      NaN      NaN  
66  산부인과전문의 1명      NaN      NaN  
67         NaN       예약      NaN  
68  산부인과전문의 2명      NaN      NaN  
69  산부인과전문의 1명       문의      NaN  

[70 rows x 8 columns]

In [ ]:
# df1= 컬럼8개 ,df2 = 컬럼8개
df3.rename(columns = {'LABEL-3':'병원명'}, inplace = True)
df3.rename(columns = {'LABEL-8':'주소'}, inplace = True)
df3.rename(columns = {'LABEL-17':'예약가능여부'}, inplace = True)
df3.rename(columns = {'LABEL-6':'진료시작시각'}, inplace = True)
df3.rename(columns = {'LABEL-10':'방문자리뷰'}, inplace = True)
df3.rename(columns = {'LABEL-11':'블로그리뷰'}, inplace = True)
df3.rename(columns = {'LABEL-12':'산부인과전문의수'}, inplace = True)
df3.rename(columns = {'LABEL-18':'기타전문의여부'}, inplace = True)
df3

병원명        진료시작시각           주소         방문자리뷰       블로그리뷰  \
0        선산부인과의원  09:00에 진료 시작   서울 관악구 신림동    방문자 리뷰 280  블로그 리뷰 102   
1     밝은미래산부인과의원  09:00에 진료 시작   서울 양천구 신정동  방문자 리뷰 1,689  블로그 리뷰 209   
2     수서나을산부인과의원  09:00에 진료 시작   서울 강남구 수서동    방문자 리뷰 363   블로그 리뷰 53   
3      한순심산부인과의원  09:30에 진료 시작   서울 강서구 화곡동    방문자 리뷰 404  블로그 리뷰 587   
4       그대의봄여성의원  09:00에 진료 시작    서울 광진구 능동    방문자 리뷰 841  블로그 리뷰 263   
..           ...           ...          ...           ...         ...   
65       봄산부인과의원  09:00에 진료 시작  서울 서대문구 홍제동    방문자 리뷰 501    블로그 리뷰 9   
66  대학로메리힐산부인과의원  10:00에 진료 시작   서울 종로구 동숭동    방문자 리뷰 733         NaN   
67   연앤네이쳐산부인과의원  10:00에 진료 시작   서울 강남구 청담동    방문자 리뷰 144  블로그 리뷰 265   
68        로즈여성의원  09:30에 진료 시작   서울 강남구 논현동     방문자 리뷰 67  블로그 리뷰 600   
69  리즈산부인과의원 홍대점  10:00에 진료 시작   서울 마포구 동교동    방문자 리뷰 396   블로그 리뷰 17   

      산부인과전문의수 예약가능여부 기타전문의여부  
0   산부인과전문의 1명    NaN     NaN  
1   산부인과전문의 4명    NaN     NaN  
2   산부인과전문의 3명    NaN     NaN  
3   산부인과전문의 1명    NaN     NaN  
4   산부인과전문의 2명    NaN     NaN  
..         ...    ...     ...  
65  산부인과전문의 1명    NaN     NaN  
66  산부인과전문의 1명    NaN     NaN  
67         NaN     예약     NaN  
68  산부인과전문의 2명    NaN     NaN  
69  산부인과전문의 1명     문의     NaN  

[70 rows x 8 columns]

In [ ]:
merged_df12

병원명           주소 예약가능여부        진료시작시각         방문자리뷰  \
0        김은주산부인과의원  서울 구로구 신도림동    NaN           NaN           NaN   
1       나를위한산부인과의원   서울 강남구 역삼동     예약           NaN           NaN   
2     애플산부인과의원 강남점   서울 서초구 서초동     예약  10:00에 진료 시작  방문자 리뷰 1,305   
3        에이린산부인과의원   서울 서초구 서초동     예약  10:00에 진료 시작    방문자 리뷰 394   
4      애플산부인과의원 잠실   서울 송파구 잠실동     예약      07/23 휴무    방문자 리뷰 958   
..             ...          ...    ...           ...           ...   
147       이사벨라여성의원   서울 강남구 대치동     예약  09:30에 진료 시작    방문자 리뷰 136   
148         우리성심의원   서울 금천구 독산동    NaN  08:00에 진료 시작    방문자 리뷰 397   
149           미즈의원   서울 은평구 불광동     예약  10:30에 진료 시작    방문자 리뷰 421   
150        솜산부인과의원   서울 관악구 남현동     예약  10:00에 진료 시작    방문자 리뷰 530   
151  헤스티아여성의원 신사본점   서울 강남구 신사동    NaN  10:00에 진료 시작    방문자 리뷰 538   

            블로그리뷰    산부인과전문의수      기타전문의여부  
0             NaN         NaN          NaN  
1             NaN         NaN          NaN  
2      블로그 리뷰 650  산부인과전문의 3명          NaN  
3    블로그 리뷰 1,353  산부인과전문의 1명          NaN  
4      블로그 리뷰 824  산부인과전문의 2명          NaN  
..            ...         ...          ...  
147    블로그 리뷰 129  산부인과전문의 1명  가정의학과전문의 1명  
148    블로그 리뷰 234  산부인과전문의 1명     내과전문의 3명  
149    블로그 리뷰 859         NaN          NaN  
150  블로그 리뷰 1,284  산부인과전문의 1명          NaN  
151    블로그 리뷰 640  산부인과전문의 6명          NaN  

[152 rows x 8 columns]

In [ ]:
#컬럼순서 맞춰주기
cols = ['병원명', '주소', '예약가능여부', '진료시작시각', '방문자리뷰', '블로그리뷰', '산부인과전문의수', '기타전문의여부']
df3 = df3[cols]
df3

병원명           주소 예약가능여부        진료시작시각         방문자리뷰       블로그리뷰  \
0        선산부인과의원   서울 관악구 신림동    NaN  09:00에 진료 시작    방문자 리뷰 280  블로그 리뷰 102   
1     밝은미래산부인과의원   서울 양천구 신정동    NaN  09:00에 진료 시작  방문자 리뷰 1,689  블로그 리뷰 209   
2     수서나을산부인과의원   서울 강남구 수서동    NaN  09:00에 진료 시작    방문자 리뷰 363   블로그 리뷰 53   
3      한순심산부인과의원   서울 강서구 화곡동    NaN  09:30에 진료 시작    방문자 리뷰 404  블로그 리뷰 587   
4       그대의봄여성의원    서울 광진구 능동    NaN  09:00에 진료 시작    방문자 리뷰 841  블로그 리뷰 263   
..           ...          ...    ...           ...           ...         ...   
65       봄산부인과의원  서울 서대문구 홍제동    NaN  09:00에 진료 시작    방문자 리뷰 501    블로그 리뷰 9   
66  대학로메리힐산부인과의원   서울 종로구 동숭동    NaN  10:00에 진료 시작    방문자 리뷰 733         NaN   
67   연앤네이쳐산부인과의원   서울 강남구 청담동     예약  10:00에 진료 시작    방문자 리뷰 144  블로그 리뷰 265   
68        로즈여성의원   서울 강남구 논현동    NaN  09:30에 진료 시작     방문자 리뷰 67  블로그 리뷰 600   
69  리즈산부인과의원 홍대점   서울 마포구 동교동     문의  10:00에 진료 시작    방문자 리뷰 396   블로그 리뷰 17   

      산부인과전문의수 기타전문의여부  
0   산부인과전문의 1명     NaN  
1   산부인과전문의 4명     NaN  
2   산부인과전문의 3명     NaN  
3   산부인과전문의 1명     NaN  
4   산부인과전문의 2명     NaN  
..         ...     ...  
65  산부인과전문의 1명     NaN  
66  산부인과전문의 1명     NaN  
67         NaN     NaN  
68  산부인과전문의 2명     NaN  
69  산부인과전문의 1명     NaN  

[70 rows x 8 columns]

## df1,2 + 3 merging

In [ ]:
# 두 데이터프레임을 병합합니다 (기본적으로 아래쪽으로 병합됨)
merged_df123 = pd.concat([merged_df12, df3], ignore_index=True)
merged_df123

병원명           주소 예약가능여부        진료시작시각         방문자리뷰  \
0       김은주산부인과의원  서울 구로구 신도림동    NaN           NaN           NaN   
1      나를위한산부인과의원   서울 강남구 역삼동     예약           NaN           NaN   
2    애플산부인과의원 강남점   서울 서초구 서초동     예약  10:00에 진료 시작  방문자 리뷰 1,305   
3       에이린산부인과의원   서울 서초구 서초동     예약  10:00에 진료 시작    방문자 리뷰 394   
4     애플산부인과의원 잠실   서울 송파구 잠실동     예약      07/23 휴무    방문자 리뷰 958   
..            ...          ...    ...           ...           ...   
217       봄산부인과의원  서울 서대문구 홍제동    NaN  09:00에 진료 시작    방문자 리뷰 501   
218  대학로메리힐산부인과의원   서울 종로구 동숭동    NaN  10:00에 진료 시작    방문자 리뷰 733   
219   연앤네이쳐산부인과의원   서울 강남구 청담동     예약  10:00에 진료 시작    방문자 리뷰 144   
220        로즈여성의원   서울 강남구 논현동    NaN  09:30에 진료 시작     방문자 리뷰 67   
221  리즈산부인과의원 홍대점   서울 마포구 동교동     문의  10:00에 진료 시작    방문자 리뷰 396   

            블로그리뷰    산부인과전문의수 기타전문의여부  
0             NaN         NaN     NaN  
1             NaN         NaN     NaN  
2      블로그 리뷰 650  산부인과전문의 3명     NaN  
3    블로그 리뷰 1,353  산부인과전문의 1명     NaN  
4      블로그 리뷰 824  산부인과전문의 2명     NaN  
..            ...         ...     ...  
217      블로그 리뷰 9  산부인과전문의 1명     NaN  
218           NaN  산부인과전문의 1명     NaN  
219    블로그 리뷰 265         NaN     NaN  
220    블로그 리뷰 600  산부인과전문의 2명     NaN  
221     블로그 리뷰 17  산부인과전문의 1명     NaN  

[222 rows x 8 columns]

In [ ]:
# 파일다운
from google.colab import files

# 변경된 데이터를 엑셀 파일로 저장합니다
merged_df123.to_excel('merged_df123.xlsx', index=False)

# 저장한 파일을 다운로드할 수 있도록 합니다
files.download('merged_df123.xlsx')

<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>

# df4 정리

- 필요컬럼: 3.6,8,10,11,12,17,20
- 필요 없는 컬럼: 1,2,4,5,7,9,13,14,15,16,18,19,

In [ ]:
#엑셀 읽어오기
df4 = pd.read_excel('./산부인과4.xlsx')
df4

LABEL-1  LABEL-2      LABEL-3  \
0   https://search.pstatic.net/common/?autoRotate=...     12.0       이로여성의원   
1   https://search.pstatic.net/common/?autoRotate=...     11.0     에펠산부인과의원   
2   https://search.pstatic.net/common/?autoRotate=...     13.0     아이엠퀸여성의원   
3   https://search.pstatic.net/common/?autoRotate=...      4.0   종로와이산부인과의원   
4   https://search.pstatic.net/common/?autoRotate=...      NaN     연세산부인과의원   
..                                                ...      ...          ...   
65                                                NaN      6.0  연세365산부인과의원   
66                                                NaN      NaN   미즈아인산부인과의원   
67                                                NaN      8.0       용훼미리의원   
68                                                NaN      NaN        무지개의원   
69                                                NaN      5.0     세화산부인과의원   

   LABEL-4 LABEL-5       LABEL-6       LABEL-7       LABEL-8  LABEL-9  \
0     산부인과   진료 종료  09:30에 진료 시작  현재 위치에서9.5km    서울 은평구 응암동  상세주소 열기   
1     산부인과   진료 종료  09:30에 진료 시작  현재 위치에서9.0km    서울 강남구 대치동  상세주소 열기   
2     산부인과   진료 종료  10:00에 진료 시작  현재 위치에서7.6km    서울 서초구 반포동  상세주소 열기   
3     산부인과   진료 종료  10:00에 진료 시작  현재 위치에서4.7km   서울 종로구 종로2가  상세주소 열기   
4     산부인과   진료 종료  09:00에 진료 시작  현재 위치에서5.3km    서울 용산구 한남동  상세주소 열기   
..     ...     ...           ...           ...           ...      ...   
65    산부인과   진료 종료  09:00에 진료 시작   현재 위치에서12km    서울 관악구 남현동  상세주소 열기   
66    산부인과   진료 종료  09:00에 진료 시작  현재 위치에서1.1km  서울 동대문구 청량리동  상세주소 열기   
67   병원,의원   진료 종료  10:00에 진료 시작  현재 위치에서6.0km   서울 용산구 이태원동  상세주소 열기   
68   병원,의원   진료 종료  09:00에 진료 시작  현재 위치에서8.8km    서울 강남구 역삼동  상세주소 열기   
69    산부인과   진료 종료  09:00에 진료 시작   현재 위치에서11km    서울 은평구 대조동  상세주소 열기   

      LABEL-10  ... LABEL-21 LABEL-22 LABEL-23 LABEL-24 LABEL-25 LABEL-26  \
0   방문자 리뷰 395  ...      NaN      NaN      NaN      NaN      NaN      NaN   
1   방문자 리뷰 417  ...      NaN      NaN      NaN      NaN      NaN      NaN   
2   방문자 리뷰 467  ...      NaN      NaN      NaN      NaN      NaN      NaN   
3   방문자 리뷰 559  ...      NaN      NaN      NaN      NaN      NaN      NaN   
4   방문자 리뷰 151  ...      NaN      NaN      NaN      NaN      NaN      NaN   
..         ...  ...      ...      ...      ...      ...      ...      ...   
65  방문자 리뷰 242  ...      NaN      NaN      NaN      NaN      NaN      NaN   
66  방문자 리뷰 226  ...      NaN      NaN      NaN      NaN      NaN      NaN   
67  방문자 리뷰 391  ...      NaN      NaN      NaN      NaN      NaN      NaN   
68  방문자 리뷰 135  ...      NaN      NaN      NaN      NaN      NaN      NaN   
69  방문자 리뷰 769  ...      NaN      NaN      NaN      NaN      NaN      NaN   

   LABEL-27 LABEL-28 LABEL-29 LABEL-30  
0       NaN      NaN      NaN      NaN  
1       NaN      NaN      NaN      NaN  
2       NaN      NaN      NaN      NaN  
3       NaN      NaN      NaN      NaN  
4       NaN      NaN      NaN      NaN  
..      ...      ...      ...      ...  
65      NaN      NaN      NaN      NaN  
66      NaN      NaN      NaN      NaN  
67      NaN      NaN      NaN      NaN  
68      NaN      NaN      NaN      NaN  
69      NaN      NaN      NaN      NaN  

[70 rows x 30 columns]

In [ ]:
df4 = df4.drop(columns = ['LABEL-1','LABEL-2','LABEL-4','LABEL-5','LABEL-7','LABEL-9','LABEL-13','LABEL-14','LABEL-15','LABEL-16'])
df4 = df4.drop(columns = ['LABEL-18','LABEL-19','LABEL-21','LABEL-22','LABEL-23','LABEL-24','LABEL-25','LABEL-26','LABEL-27','LABEL-28','LABEL-29','LABEL-30'])
df4

LABEL-3       LABEL-6       LABEL-8    LABEL-10      LABEL-11  \
0        이로여성의원  09:30에 진료 시작    서울 은평구 응암동  방문자 리뷰 395  블로그 리뷰 1,056   
1      에펠산부인과의원  09:30에 진료 시작    서울 강남구 대치동  방문자 리뷰 417    블로그 리뷰 334   
2      아이엠퀸여성의원  10:00에 진료 시작    서울 서초구 반포동  방문자 리뷰 467    블로그 리뷰 522   
3    종로와이산부인과의원  10:00에 진료 시작   서울 종로구 종로2가  방문자 리뷰 559    블로그 리뷰 971   
4      연세산부인과의원  09:00에 진료 시작    서울 용산구 한남동  방문자 리뷰 151      블로그 리뷰 5   
..          ...           ...           ...         ...           ...   
65  연세365산부인과의원  09:00에 진료 시작    서울 관악구 남현동  방문자 리뷰 242     블로그 리뷰 53   
66   미즈아인산부인과의원  09:00에 진료 시작  서울 동대문구 청량리동  방문자 리뷰 226      블로그 리뷰 5   
67       용훼미리의원  10:00에 진료 시작   서울 용산구 이태원동  방문자 리뷰 391      블로그 리뷰 2   
68        무지개의원  09:00에 진료 시작    서울 강남구 역삼동  방문자 리뷰 135      블로그 리뷰 9   
69     세화산부인과의원  09:00에 진료 시작    서울 은평구 대조동  방문자 리뷰 769    블로그 리뷰 306   

       LABEL-12 LABEL-17 LABEL-20  
0    산부인과전문의 1명      NaN      NaN  
1    산부인과전문의 1명       예약      NaN  
2    산부인과전문의 1명       예약      NaN  
3    산부인과전문의 1명      NaN      NaN  
4    산부인과전문의 1명      NaN      NaN  
..          ...      ...      ...  
65   산부인과전문의 1명      NaN      NaN  
66   산부인과전문의 1명      NaN      NaN  
67  가정의학과전문의 1명      NaN      NaN  
68          NaN      NaN      NaN  
69   산부인과전문의 1명       예약      NaN  

[70 rows x 8 columns]

In [ ]:
# df1= 컬럼8개 ,df2 = 컬럼8개
df4.rename(columns = {'LABEL-3':'병원명'}, inplace = True)
df4.rename(columns = {'LABEL-8':'주소'}, inplace = True)
df4.rename(columns = {'LABEL-17':'예약가능여부'}, inplace = True)
df4.rename(columns = {'LABEL-6':'진료시작시각'}, inplace = True)
df4.rename(columns = {'LABEL-10':'방문자리뷰'}, inplace = True)
df4.rename(columns = {'LABEL-11':'블로그리뷰'}, inplace = True)
df4.rename(columns = {'LABEL-12':'산부인과전문의수'}, inplace = True)
df4.rename(columns = {'LABEL-20':'기타전문의여부'}, inplace = True)
df4

병원명        진료시작시각            주소       방문자리뷰         블로그리뷰  \
0        이로여성의원  09:30에 진료 시작    서울 은평구 응암동  방문자 리뷰 395  블로그 리뷰 1,056   
1      에펠산부인과의원  09:30에 진료 시작    서울 강남구 대치동  방문자 리뷰 417    블로그 리뷰 334   
2      아이엠퀸여성의원  10:00에 진료 시작    서울 서초구 반포동  방문자 리뷰 467    블로그 리뷰 522   
3    종로와이산부인과의원  10:00에 진료 시작   서울 종로구 종로2가  방문자 리뷰 559    블로그 리뷰 971   
4      연세산부인과의원  09:00에 진료 시작    서울 용산구 한남동  방문자 리뷰 151      블로그 리뷰 5   
..          ...           ...           ...         ...           ...   
65  연세365산부인과의원  09:00에 진료 시작    서울 관악구 남현동  방문자 리뷰 242     블로그 리뷰 53   
66   미즈아인산부인과의원  09:00에 진료 시작  서울 동대문구 청량리동  방문자 리뷰 226      블로그 리뷰 5   
67       용훼미리의원  10:00에 진료 시작   서울 용산구 이태원동  방문자 리뷰 391      블로그 리뷰 2   
68        무지개의원  09:00에 진료 시작    서울 강남구 역삼동  방문자 리뷰 135      블로그 리뷰 9   
69     세화산부인과의원  09:00에 진료 시작    서울 은평구 대조동  방문자 리뷰 769    블로그 리뷰 306   

       산부인과전문의수 예약가능여부 기타전문의여부  
0    산부인과전문의 1명    NaN     NaN  
1    산부인과전문의 1명     예약     NaN  
2    산부인과전문의 1명     예약     NaN  
3    산부인과전문의 1명    NaN     NaN  
4    산부인과전문의 1명    NaN     NaN  
..          ...    ...     ...  
65   산부인과전문의 1명    NaN     NaN  
66   산부인과전문의 1명    NaN     NaN  
67  가정의학과전문의 1명    NaN     NaN  
68          NaN    NaN     NaN  
69   산부인과전문의 1명     예약     NaN  

[70 rows x 8 columns]

In [ ]:
#컬럼순서 맞춰주기
cols = ['병원명', '주소', '예약가능여부', '진료시작시각', '방문자리뷰', '블로그리뷰', '산부인과전문의수', '기타전문의여부']
df4 = df4[cols]
df4

병원명            주소 예약가능여부        진료시작시각       방문자리뷰         블로그리뷰  \
0        이로여성의원    서울 은평구 응암동    NaN  09:30에 진료 시작  방문자 리뷰 395  블로그 리뷰 1,056   
1      에펠산부인과의원    서울 강남구 대치동     예약  09:30에 진료 시작  방문자 리뷰 417    블로그 리뷰 334   
2      아이엠퀸여성의원    서울 서초구 반포동     예약  10:00에 진료 시작  방문자 리뷰 467    블로그 리뷰 522   
3    종로와이산부인과의원   서울 종로구 종로2가    NaN  10:00에 진료 시작  방문자 리뷰 559    블로그 리뷰 971   
4      연세산부인과의원    서울 용산구 한남동    NaN  09:00에 진료 시작  방문자 리뷰 151      블로그 리뷰 5   
..          ...           ...    ...           ...         ...           ...   
65  연세365산부인과의원    서울 관악구 남현동    NaN  09:00에 진료 시작  방문자 리뷰 242     블로그 리뷰 53   
66   미즈아인산부인과의원  서울 동대문구 청량리동    NaN  09:00에 진료 시작  방문자 리뷰 226      블로그 리뷰 5   
67       용훼미리의원   서울 용산구 이태원동    NaN  10:00에 진료 시작  방문자 리뷰 391      블로그 리뷰 2   
68        무지개의원    서울 강남구 역삼동    NaN  09:00에 진료 시작  방문자 리뷰 135      블로그 리뷰 9   
69     세화산부인과의원    서울 은평구 대조동     예약  09:00에 진료 시작  방문자 리뷰 769    블로그 리뷰 306   

       산부인과전문의수 기타전문의여부  
0    산부인과전문의 1명     NaN  
1    산부인과전문의 1명     NaN  
2    산부인과전문의 1명     NaN  
3    산부인과전문의 1명     NaN  
4    산부인과전문의 1명     NaN  
..          ...     ...  
65   산부인과전문의 1명     NaN  
66   산부인과전문의 1명     NaN  
67  가정의학과전문의 1명     NaN  
68          NaN     NaN  
69   산부인과전문의 1명     NaN  

[70 rows x 8 columns]

## merged_df123 + 4 병합

In [ ]:
# 두 데이터프레임을 병합합니다 (기본적으로 아래쪽으로 병합됨)
merged_df1234 = pd.concat([merged_df123, df4], ignore_index=True)
merged_df1234

병원명            주소 예약가능여부        진료시작시각         방문자리뷰  \
0       김은주산부인과의원   서울 구로구 신도림동    NaN           NaN           NaN   
1      나를위한산부인과의원    서울 강남구 역삼동     예약           NaN           NaN   
2    애플산부인과의원 강남점    서울 서초구 서초동     예약  10:00에 진료 시작  방문자 리뷰 1,305   
3       에이린산부인과의원    서울 서초구 서초동     예약  10:00에 진료 시작    방문자 리뷰 394   
4     애플산부인과의원 잠실    서울 송파구 잠실동     예약      07/23 휴무    방문자 리뷰 958   
..            ...           ...    ...           ...           ...   
287   연세365산부인과의원    서울 관악구 남현동    NaN  09:00에 진료 시작    방문자 리뷰 242   
288    미즈아인산부인과의원  서울 동대문구 청량리동    NaN  09:00에 진료 시작    방문자 리뷰 226   
289        용훼미리의원   서울 용산구 이태원동    NaN  10:00에 진료 시작    방문자 리뷰 391   
290         무지개의원    서울 강남구 역삼동    NaN  09:00에 진료 시작    방문자 리뷰 135   
291      세화산부인과의원    서울 은평구 대조동     예약  09:00에 진료 시작    방문자 리뷰 769   

            블로그리뷰     산부인과전문의수 기타전문의여부  
0             NaN          NaN     NaN  
1             NaN          NaN     NaN  
2      블로그 리뷰 650   산부인과전문의 3명     NaN  
3    블로그 리뷰 1,353   산부인과전문의 1명     NaN  
4      블로그 리뷰 824   산부인과전문의 2명     NaN  
..            ...          ...     ...  
287     블로그 리뷰 53   산부인과전문의 1명     NaN  
288      블로그 리뷰 5   산부인과전문의 1명     NaN  
289      블로그 리뷰 2  가정의학과전문의 1명     NaN  
290      블로그 리뷰 9          NaN     NaN  
291    블로그 리뷰 306   산부인과전문의 1명     NaN  

[292 rows x 8 columns]

In [ ]:
# 파일다운
from google.colab import files

# 변경된 데이터를 엑셀 파일로 저장합니다
merged_df123.to_excel('merged_df1234.xlsx', index=False)

# 저장한 파일을 다운로드할 수 있도록 합니다
files.download('merged_df1234.xlsx')

<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>

# df5
- 필요한 컬럼:17,20,22,24,25,26,28
- 필요없는 컬럼:1~16,18,19,21,23,27,29
- 유의: 기타 전문의가 명시되어있는 병원이 없어 아예 행이 없는 상태 -> Nan넣어
추가해줌

In [ ]:
df5 = pd.read_excel('./산부인과5.xlsx')
df5

LABEL-1  LABEL-2  LABEL-3  LABEL-4  LABEL-5  LABEL-6  LABEL-7  LABEL-8  \
0       NaN      NaN      NaN      NaN      NaN      NaN      NaN      NaN   
1       NaN      NaN      NaN      NaN      NaN      NaN      NaN      NaN   
2       NaN      NaN      NaN      NaN      NaN      NaN      NaN      NaN   
3       NaN      NaN      NaN      NaN      NaN      NaN      NaN      NaN   
4       NaN      NaN      NaN      NaN      NaN      NaN      NaN      NaN   
5       NaN      NaN      NaN      NaN      NaN      NaN      NaN      NaN   
6       NaN      NaN      NaN      NaN      NaN      NaN      NaN      NaN   
7       NaN      NaN      NaN      NaN      NaN      NaN      NaN      NaN   
8       NaN      NaN      NaN      NaN      NaN      NaN      NaN      NaN   
9       NaN      NaN      NaN      NaN      NaN      NaN      NaN      NaN   
10      NaN      NaN      NaN      NaN      NaN      NaN      NaN      NaN   
11      NaN      NaN      NaN      NaN      NaN      NaN      NaN      NaN   
12      NaN      NaN      NaN      NaN      NaN      NaN      NaN      NaN   
13      NaN      NaN      NaN      NaN      NaN      NaN      NaN      NaN   
14      NaN      NaN      NaN      NaN      NaN      NaN      NaN      NaN   
15      NaN      NaN      NaN      NaN      NaN      NaN      NaN      NaN   
16      NaN      NaN      NaN      NaN      NaN      NaN      NaN      NaN   
17      NaN      NaN      NaN      NaN      NaN      NaN      NaN      NaN   
18      NaN      NaN      NaN      NaN      NaN      NaN      NaN      NaN   
19      NaN      NaN      NaN      NaN      NaN      NaN      NaN      NaN   

    LABEL-9  LABEL-10  ...      LABEL-20      LABEL-21       LABEL-22  \
0       NaN       NaN  ...  09:00에 진료 시작  현재 위치에서9.9km     서울 도봉구 방학동   
1       NaN       NaN  ...  09:00에 진료 시작   현재 위치에서17km     서울 금천구 가산동   
2       NaN       NaN  ...  09:30에 진료 시작  현재 위치에서9.0km     서울 도봉구 쌍문동   
3       NaN       NaN  ...  09:00에 진료 시작  현재 위치에서9.9km     서울 마포구 동교동   
4       NaN       NaN  ...  09:00에 진료 시작   현재 위치에서10km     서울 노원구 상계동   
5       NaN       NaN  ...  10:00에 진료 시작  현재 위치에서7.9km     서울 강남구 논현동   
6       NaN       NaN  ...  10:00에 진료 시작   현재 위치에서12km    서울 동작구 노량진동   
7       NaN       NaN  ...  09:30에 진료 시작  현재 위치에서3.0km    서울 성북구 하월곡동   
8       NaN       NaN  ...  10:00에 진료 시작  현재 위치에서7.6km     서울 마포구 아현동   
9       NaN       NaN  ...  09:30에 진료 시작   현재 위치에서18km     서울 금천구 시흥동   
10      NaN       NaN  ...  09:30에 진료 시작  현재 위치에서6.3km      서울 종로구 평동   
11      NaN       NaN  ...  10:00에 진료 시작  현재 위치에서4.2km     서울 광진구 화양동   
12      NaN       NaN  ...  09:30에 진료 시작  현재 위치에서7.9km     서울 강남구 대치동   
13      NaN       NaN  ...  10:00에 진료 시작  현재 위치에서8.2km     서울 송파구 잠실동   
14      NaN       NaN  ...  09:00에 진료 시작   현재 위치에서13km    서울 영등포구 신길동   
15      NaN       NaN  ...  09:30에 진료 시작   현재 위치에서11km   서울 서대문구 북가좌동   
16      NaN       NaN  ...  10:00에 진료 시작  현재 위치에서2.6km  서울 성북구 동소문동5가   
17      NaN       NaN  ...  09:30에 진료 시작   현재 위치에서14km      서울 양천구 목동   
18      NaN       NaN  ...  10:00에 진료 시작   현재 위치에서15km      서울 양천구 목동   
19      NaN       NaN  ...  09:30에 진료 시작  현재 위치에서8.7km     서울 송파구 신천동   

   LABEL-23    LABEL-24      LABEL-25    LABEL-26  \
0   상세주소 열기  방문자 리뷰 993      블로그 리뷰 3  산부인과전문의 1명   
1   상세주소 열기  방문자 리뷰 291    블로그 리뷰 650  산부인과전문의 1명   
2   상세주소 열기  방문자 리뷰 421      블로그 리뷰 3  산부인과전문의 1명   
3   상세주소 열기  방문자 리뷰 339    블로그 리뷰 141  산부인과전문의 1명   
4   상세주소 열기  방문자 리뷰 620      블로그 리뷰 2  산부인과전문의 1명   
5   상세주소 열기   방문자 리뷰 88      블로그 리뷰 4  산부인과전문의 1명   
6   상세주소 열기   방문자 리뷰 63      블로그 리뷰 2  산부인과전문의 1명   
7   상세주소 열기  방문자 리뷰 400      블로그 리뷰 4  산부인과전문의 1명   
8   상세주소 열기  방문자 리뷰 205      블로그 리뷰 3  산부인과전문의 1명   
9   상세주소 열기  방문자 리뷰 356      블로그 리뷰 4  산부인과전문의 1명   
10  상세주소 열기  방문자 리뷰 250     블로그 리뷰 13  산부인과전문의 1명   
11  상세주소 열기  방문자 리뷰 347      블로그 리뷰 6  산부인과전문의 1명   
12  상세주소 열기  방문자 리뷰 144      블로그 리뷰 1  산부인과전문의 1명   
13  상세주소 열기  방문자 리뷰 315    블로그 리뷰 432  산부인과전문의 1명   
14  상세주소 열기  방문자 리뷰 385    블로그 리뷰 12

In [ ]:
for i in range(1,17):
  df5 = df5.drop(columns = ['LABEL-'+str(i)])
df5

LABEL-17 LABEL-18 LABEL-19      LABEL-20      LABEL-21       LABEL-22  \
0       혜정산부인과의원     산부인과    진료 종료  09:00에 진료 시작  현재 위치에서9.9km     서울 도봉구 방학동   
1       웨일산부인과의원     산부인과    진료 종료  09:00에 진료 시작   현재 위치에서17km     서울 금천구 가산동   
2      왕영미산부인과의원     산부인과    진료 종료  09:30에 진료 시작  현재 위치에서9.0km     서울 도봉구 쌍문동   
3      진오비산부인과의원     산부인과    진료 종료  09:00에 진료 시작  현재 위치에서9.9km     서울 마포구 동교동   
4      박규희산부인과의원     산부인과    진료 종료  09:00에 진료 시작   현재 위치에서10km     서울 노원구 상계동   
5     프린세스산부인과의원     산부인과    진료 종료  10:00에 진료 시작  현재 위치에서7.9km     서울 강남구 논현동   
6          색다른의원     산부인과    진료 종료  10:00에 진료 시작   현재 위치에서12km    서울 동작구 노량진동   
7       제니산부인과의원     산부인과    진료 종료  09:30에 진료 시작  현재 위치에서3.0km    서울 성북구 하월곡동   
8    이화더블유산부인과의원     산부인과    진료 종료  10:00에 진료 시작  현재 위치에서7.6km     서울 마포구 아현동   
9      이성림산부인과의원     산부인과    진료 종료  09:30에 진료 시작   현재 위치에서18km     서울 금천구 시흥동   
10    연세조이산부인과의원     산부인과    진료 종료  09:30에 진료 시작  현재 위치에서6.3km      서울 종로구 평동   
11       김숙희여성의원     산부인과    진료 종료  10:00에 진료 시작  현재 위치에서4.2km     서울 광진구 화양동   
12    삼성연세산부인과의원     산부인과    진료 종료  09:30에 진료 시작  현재 위치에서7.9km     서울 강남구 대치동   
13     김윤숙산부인과의원     산부인과    진료 종료  10:00에 진료 시작  현재 위치에서8.2km     서울 송파구 잠실동   
14     라메르산부인과의원     산부인과    진료 종료  09:00에 진료 시작   현재 위치에서13km    서울 영등포구 신길동   
15      봄꽃산부인과의원     산부인과    진료 종료  09:30에 진료 시작   현재 위치에서11km   서울 서대문구 북가좌동   
16    더행복한산부인과의원     산부인과    진료 종료  10:00에 진료 시작  현재 위치에서2.6km  서울 성북구 동소문동5가   
17     여미애산부인과의원     산부인과    진료 종료  09:30에 진료 시작   현재 위치에서14km      서울 양천구 목동   
18        우디여성의원     산부인과    진료 종료  10:00에 진료 시작   현재 위치에서15km      서울 양천구 목동   
19  김성희마리안산부인과의원     산부인과    진료 종료  09:30에 진료 시작  현재 위치에서8.7km     서울 송파구 신천동   

   LABEL-23    LABEL-24      LABEL-25    LABEL-26  \
0   상세주소 열기  방문자 리뷰 993      블로그 리뷰 3  산부인과전문의 1명   
1   상세주소 열기  방문자 리뷰 291    블로그 리뷰 650  산부인과전문의 1명   
2   상세주소 열기  방문자 리뷰 421      블로그 리뷰 3  산부인과전문의 1명   
3   상세주소 열기  방문자 리뷰 339    블로그 리뷰 141  산부인과전문의 1명   
4   상세주소 열기  방문자 리뷰 620      블로그 리뷰 2  산부인과전문의 1명   
5   상세주소 열기   방문자 리뷰 88      블로그 리뷰 4  산부인과전문의 1명   
6   상세주소 열기   방문자 리뷰 63      블로그 리뷰 2  산부인과전문의 1명   
7   상세주소 열기  방문자 리뷰 400      블로그 리뷰 4  산부인과전문의 1명   
8   상세주소 열기  방문자 리뷰 205      블로그 리뷰 3  산부인과전문의 1명   
9   상세주소 열기  방문자 리뷰 356      블로그 리뷰 4  산부인과전문의 1명   
10  상세주소 열기  방문자 리뷰 250     블로그 리뷰 13  산부인과전문의 1명   
11  상세주소 열기  방문자 리뷰 347      블로그 리뷰 6  산부인과전문의 1명   
12  상세주소 열기  방문자 리뷰 144      블로그 리뷰 1  산부인과전문의 1명   
13  상세주소 열기  방문자 리뷰 315    블로그 리뷰 432  산부인과전문의 1명   
14  상세주소 열기  방문자 리뷰 385    블로그 리뷰 126  산부인과전문의 1명   
15  상세주소 열기   방문자 리뷰 26     블로그 리뷰 24  산부인과전문의 1명   
16  상세주소 열기  방문자 리뷰 508    블로그 리뷰 664         NaN   
17  상세주소 열기  방문자 리뷰 427  블로그 리뷰 1,154  산부인과전문의 1명   
18  상세주소 열기  방문자 리뷰 321      블로그 리뷰 2  산부인과전문의 1명   
19  상세주소 열기  방문자 리뷰 284      블로그 리뷰 3  산부인과전문의 1명   

                                             LABEL-27 LABEL-28 LABEL-29  
0                                                 NaN      NaN      NaN  
1                                                 NaN      NaN      NaN  
2                                                 NaN      NaN      NaN  
3   https://pcmap.place.naver.com/hospital/list?qu...      NaN      NaN  
4                                                 NaN      NaN      NaN  
5                                                 NaN       예약       문의  
6                                                 NaN      NaN      NaN  
7                                                 NaN      NaN      NaN  
8   https://pcmap.place.naver.com/hospital/list?qu...      NaN      NaN  
9                                                 NaN      NaN      NaN  
10                                                NaN      NaN      NaN  
11                                                NaN      NaN      NaN  
12                                                NaN      NaN      NaN  
13                                                NaN       문의      NaN  
14                                                NaN      NaN      NaN  
15      

In [ ]:
# 18,19,21,23,27,29
df5 = df5.drop(columns = ['LABEL-18','LABEL-19','LABEL-21','LABEL-23','LABEL-27','LABEL-29'])
df5

LABEL-17      LABEL-20       LABEL-22    LABEL-24      LABEL-25  \
0       혜정산부인과의원  09:00에 진료 시작     서울 도봉구 방학동  방문자 리뷰 993      블로그 리뷰 3   
1       웨일산부인과의원  09:00에 진료 시작     서울 금천구 가산동  방문자 리뷰 291    블로그 리뷰 650   
2      왕영미산부인과의원  09:30에 진료 시작     서울 도봉구 쌍문동  방문자 리뷰 421      블로그 리뷰 3   
3      진오비산부인과의원  09:00에 진료 시작     서울 마포구 동교동  방문자 리뷰 339    블로그 리뷰 141   
4      박규희산부인과의원  09:00에 진료 시작     서울 노원구 상계동  방문자 리뷰 620      블로그 리뷰 2   
5     프린세스산부인과의원  10:00에 진료 시작     서울 강남구 논현동   방문자 리뷰 88      블로그 리뷰 4   
6          색다른의원  10:00에 진료 시작    서울 동작구 노량진동   방문자 리뷰 63      블로그 리뷰 2   
7       제니산부인과의원  09:30에 진료 시작    서울 성북구 하월곡동  방문자 리뷰 400      블로그 리뷰 4   
8    이화더블유산부인과의원  10:00에 진료 시작     서울 마포구 아현동  방문자 리뷰 205      블로그 리뷰 3   
9      이성림산부인과의원  09:30에 진료 시작     서울 금천구 시흥동  방문자 리뷰 356      블로그 리뷰 4   
10    연세조이산부인과의원  09:30에 진료 시작      서울 종로구 평동  방문자 리뷰 250     블로그 리뷰 13   
11       김숙희여성의원  10:00에 진료 시작     서울 광진구 화양동  방문자 리뷰 347      블로그 리뷰 6   
12    삼성연세산부인과의원  09:30에 진료 시작     서울 강남구 대치동  방문자 리뷰 144      블로그 리뷰 1   
13     김윤숙산부인과의원  10:00에 진료 시작     서울 송파구 잠실동  방문자 리뷰 315    블로그 리뷰 432   
14     라메르산부인과의원  09:00에 진료 시작    서울 영등포구 신길동  방문자 리뷰 385    블로그 리뷰 126   
15      봄꽃산부인과의원  09:30에 진료 시작   서울 서대문구 북가좌동   방문자 리뷰 26     블로그 리뷰 24   
16    더행복한산부인과의원  10:00에 진료 시작  서울 성북구 동소문동5가  방문자 리뷰 508    블로그 리뷰 664   
17     여미애산부인과의원  09:30에 진료 시작      서울 양천구 목동  방문자 리뷰 427  블로그 리뷰 1,154   
18        우디여성의원  10:00에 진료 시작      서울 양천구 목동  방문자 리뷰 321      블로그 리뷰 2   
19  김성희마리안산부인과의원  09:30에 진료 시작     서울 송파구 신천동  방문자 리뷰 284      블로그 리뷰 3   

      LABEL-26 LABEL-28  
0   산부인과전문의 1명      NaN  
1   산부인과전문의 1명      NaN  
2   산부인과전문의 1명      NaN  
3   산부인과전문의 1명      NaN  
4   산부인과전문의 1명      NaN  
5   산부인과전문의 1명       예약  
6   산부인과전문의 1명      NaN  
7   산부인과전문의 1명      NaN  
8   산부인과전문의 1명      NaN  
9   산부인과전문의 1명      NaN  
10  산부인과전문의 1명      NaN  
11  산부인과전문의 1명      NaN  
12  산부인과전문의 1명      NaN  
13  산부인과전문의 1명       문의  
14  산부인과전문의 1명      NaN  
15  산부인과전문의 1명       예약  
16         NaN      NaN  
17  산부인과전문의 1명       예약  
18  산부인과전문의 1명      NaN  
19  산부인과전문의 1명       예약

In [ ]:
# 기타전문의 행 따로 만들어주기
import numpy as np
df5['기타전문의여부'] = np.nan
df5

LABEL-17      LABEL-20       LABEL-22    LABEL-24      LABEL-25  \
0       혜정산부인과의원  09:00에 진료 시작     서울 도봉구 방학동  방문자 리뷰 993      블로그 리뷰 3   
1       웨일산부인과의원  09:00에 진료 시작     서울 금천구 가산동  방문자 리뷰 291    블로그 리뷰 650   
2      왕영미산부인과의원  09:30에 진료 시작     서울 도봉구 쌍문동  방문자 리뷰 421      블로그 리뷰 3   
3      진오비산부인과의원  09:00에 진료 시작     서울 마포구 동교동  방문자 리뷰 339    블로그 리뷰 141   
4      박규희산부인과의원  09:00에 진료 시작     서울 노원구 상계동  방문자 리뷰 620      블로그 리뷰 2   
5     프린세스산부인과의원  10:00에 진료 시작     서울 강남구 논현동   방문자 리뷰 88      블로그 리뷰 4   
6          색다른의원  10:00에 진료 시작    서울 동작구 노량진동   방문자 리뷰 63      블로그 리뷰 2   
7       제니산부인과의원  09:30에 진료 시작    서울 성북구 하월곡동  방문자 리뷰 400      블로그 리뷰 4   
8    이화더블유산부인과의원  10:00에 진료 시작     서울 마포구 아현동  방문자 리뷰 205      블로그 리뷰 3   
9      이성림산부인과의원  09:30에 진료 시작     서울 금천구 시흥동  방문자 리뷰 356      블로그 리뷰 4   
10    연세조이산부인과의원  09:30에 진료 시작      서울 종로구 평동  방문자 리뷰 250     블로그 리뷰 13   
11       김숙희여성의원  10:00에 진료 시작     서울 광진구 화양동  방문자 리뷰 347      블로그 리뷰 6   
12    삼성연세산부인과의원  09:30에 진료 시작     서울 강남구 대치동  방문자 리뷰 144      블로그 리뷰 1   
13     김윤숙산부인과의원  10:00에 진료 시작     서울 송파구 잠실동  방문자 리뷰 315    블로그 리뷰 432   
14     라메르산부인과의원  09:00에 진료 시작    서울 영등포구 신길동  방문자 리뷰 385    블로그 리뷰 126   
15      봄꽃산부인과의원  09:30에 진료 시작   서울 서대문구 북가좌동   방문자 리뷰 26     블로그 리뷰 24   
16    더행복한산부인과의원  10:00에 진료 시작  서울 성북구 동소문동5가  방문자 리뷰 508    블로그 리뷰 664   
17     여미애산부인과의원  09:30에 진료 시작      서울 양천구 목동  방문자 리뷰 427  블로그 리뷰 1,154   
18        우디여성의원  10:00에 진료 시작      서울 양천구 목동  방문자 리뷰 321      블로그 리뷰 2   
19  김성희마리안산부인과의원  09:30에 진료 시작     서울 송파구 신천동  방문자 리뷰 284      블로그 리뷰 3   

      LABEL-26 LABEL-28  기타전문의여부  
0   산부인과전문의 1명      NaN      NaN  
1   산부인과전문의 1명      NaN      NaN  
2   산부인과전문의 1명      NaN      NaN  
3   산부인과전문의 1명      NaN      NaN  
4   산부인과전문의 1명      NaN      NaN  
5   산부인과전문의 1명       예약      NaN  
6   산부인과전문의 1명      NaN      NaN  
7   산부인과전문의 1명      NaN      NaN  
8   산부인과전문의 1명      NaN      NaN  
9   산부인과전문의 1명      NaN      NaN  
10  산부인과전문의 1명      NaN      NaN  
11  산부인과전문의 1명      NaN      NaN  
12  산부인과전문의 1명      NaN      NaN  
13  산부인과전문의 1명       문의      NaN  
14  산부인과전문의 1명      NaN      NaN  
15  산부인과전문의 1명       예약      NaN  
16         NaN      NaN      NaN  
17  산부인과전문의 1명       예약      NaN  
18  산부인과전문의 1명      NaN      NaN  
19  산부인과전문의 1명       예약      NaN

In [ ]:
#행이름 정리
# df1= 컬럼8개 ,df2 = 컬럼8개
df5.rename(columns = {'LABEL-17':'병원명'}, inplace = True)
df5.rename(columns = {'LABEL-22':'주소'}, inplace = True)
df5.rename(columns = {'LABEL-28':'예약가능여부'}, inplace = True)
df5.rename(columns = {'LABEL-20':'진료시작시각'}, inplace = True)
df5.rename(columns = {'LABEL-24':'방문자리뷰'}, inplace = True)
df5.rename(columns = {'LABEL-25':'블로그리뷰'}, inplace = True)
df5.rename(columns = {'LABEL-26':'산부인과전문의수'}, inplace = True)
#df4.rename(columns = {'LABEL-20':'기타전문의여부'}, inplace = True)
df5

병원명        진료시작시각             주소       방문자리뷰         블로그리뷰  \
0       혜정산부인과의원  09:00에 진료 시작     서울 도봉구 방학동  방문자 리뷰 993      블로그 리뷰 3   
1       웨일산부인과의원  09:00에 진료 시작     서울 금천구 가산동  방문자 리뷰 291    블로그 리뷰 650   
2      왕영미산부인과의원  09:30에 진료 시작     서울 도봉구 쌍문동  방문자 리뷰 421      블로그 리뷰 3   
3      진오비산부인과의원  09:00에 진료 시작     서울 마포구 동교동  방문자 리뷰 339    블로그 리뷰 141   
4      박규희산부인과의원  09:00에 진료 시작     서울 노원구 상계동  방문자 리뷰 620      블로그 리뷰 2   
5     프린세스산부인과의원  10:00에 진료 시작     서울 강남구 논현동   방문자 리뷰 88      블로그 리뷰 4   
6          색다른의원  10:00에 진료 시작    서울 동작구 노량진동   방문자 리뷰 63      블로그 리뷰 2   
7       제니산부인과의원  09:30에 진료 시작    서울 성북구 하월곡동  방문자 리뷰 400      블로그 리뷰 4   
8    이화더블유산부인과의원  10:00에 진료 시작     서울 마포구 아현동  방문자 리뷰 205      블로그 리뷰 3   
9      이성림산부인과의원  09:30에 진료 시작     서울 금천구 시흥동  방문자 리뷰 356      블로그 리뷰 4   
10    연세조이산부인과의원  09:30에 진료 시작      서울 종로구 평동  방문자 리뷰 250     블로그 리뷰 13   
11       김숙희여성의원  10:00에 진료 시작     서울 광진구 화양동  방문자 리뷰 347      블로그 리뷰 6   
12    삼성연세산부인과의원  09:30에 진료 시작     서울 강남구 대치동  방문자 리뷰 144      블로그 리뷰 1   
13     김윤숙산부인과의원  10:00에 진료 시작     서울 송파구 잠실동  방문자 리뷰 315    블로그 리뷰 432   
14     라메르산부인과의원  09:00에 진료 시작    서울 영등포구 신길동  방문자 리뷰 385    블로그 리뷰 126   
15      봄꽃산부인과의원  09:30에 진료 시작   서울 서대문구 북가좌동   방문자 리뷰 26     블로그 리뷰 24   
16    더행복한산부인과의원  10:00에 진료 시작  서울 성북구 동소문동5가  방문자 리뷰 508    블로그 리뷰 664   
17     여미애산부인과의원  09:30에 진료 시작      서울 양천구 목동  방문자 리뷰 427  블로그 리뷰 1,154   
18        우디여성의원  10:00에 진료 시작      서울 양천구 목동  방문자 리뷰 321      블로그 리뷰 2   
19  김성희마리안산부인과의원  09:30에 진료 시작     서울 송파구 신천동  방문자 리뷰 284      블로그 리뷰 3   

      산부인과전문의수 예약가능여부  기타전문의여부  
0   산부인과전문의 1명    NaN      NaN  
1   산부인과전문의 1명    NaN      NaN  
2   산부인과전문의 1명    NaN      NaN  
3   산부인과전문의 1명    NaN      NaN  
4   산부인과전문의 1명    NaN      NaN  
5   산부인과전문의 1명     예약      NaN  
6   산부인과전문의 1명    NaN      NaN  
7   산부인과전문의 1명    NaN      NaN  
8   산부인과전문의 1명    NaN      NaN  
9   산부인과전문의 1명    NaN      NaN  
10  산부인과전문의 1명    NaN      NaN  
11  산부인과전문의 1명    NaN      NaN  
12  산부인과전문의 1명    NaN      NaN  
13  산부인과전문의 1명     문의      NaN  
14  산부인과전문의 1명    NaN      NaN  
15  산부인과전문의 1명     예약      NaN  
16         NaN    NaN      NaN  
17  산부인과전문의 1명     예약      NaN  
18  산부인과전문의 1명    NaN      NaN  
19  산부인과전문의 1명     예약      NaN

In [ ]:
#컬럼순서 맞춰주기
cols = ['병원명', '주소', '예약가능여부', '진료시작시각', '방문자리뷰', '블로그리뷰', '산부인과전문의수', '기타전문의여부']
df5 = df5[cols]
df5

병원명             주소 예약가능여부        진료시작시각       방문자리뷰  \
0       혜정산부인과의원     서울 도봉구 방학동    NaN  09:00에 진료 시작  방문자 리뷰 993   
1       웨일산부인과의원     서울 금천구 가산동    NaN  09:00에 진료 시작  방문자 리뷰 291   
2      왕영미산부인과의원     서울 도봉구 쌍문동    NaN  09:30에 진료 시작  방문자 리뷰 421   
3      진오비산부인과의원     서울 마포구 동교동    NaN  09:00에 진료 시작  방문자 리뷰 339   
4      박규희산부인과의원     서울 노원구 상계동    NaN  09:00에 진료 시작  방문자 리뷰 620   
5     프린세스산부인과의원     서울 강남구 논현동     예약  10:00에 진료 시작   방문자 리뷰 88   
6          색다른의원    서울 동작구 노량진동    NaN  10:00에 진료 시작   방문자 리뷰 63   
7       제니산부인과의원    서울 성북구 하월곡동    NaN  09:30에 진료 시작  방문자 리뷰 400   
8    이화더블유산부인과의원     서울 마포구 아현동    NaN  10:00에 진료 시작  방문자 리뷰 205   
9      이성림산부인과의원     서울 금천구 시흥동    NaN  09:30에 진료 시작  방문자 리뷰 356   
10    연세조이산부인과의원      서울 종로구 평동    NaN  09:30에 진료 시작  방문자 리뷰 250   
11       김숙희여성의원     서울 광진구 화양동    NaN  10:00에 진료 시작  방문자 리뷰 347   
12    삼성연세산부인과의원     서울 강남구 대치동    NaN  09:30에 진료 시작  방문자 리뷰 144   
13     김윤숙산부인과의원     서울 송파구 잠실동     문의  10:00에 진료 시작  방문자 리뷰 315   
14     라메르산부인과의원    서울 영등포구 신길동    NaN  09:00에 진료 시작  방문자 리뷰 385   
15      봄꽃산부인과의원   서울 서대문구 북가좌동     예약  09:30에 진료 시작   방문자 리뷰 26   
16    더행복한산부인과의원  서울 성북구 동소문동5가    NaN  10:00에 진료 시작  방문자 리뷰 508   
17     여미애산부인과의원      서울 양천구 목동     예약  09:30에 진료 시작  방문자 리뷰 427   
18        우디여성의원      서울 양천구 목동    NaN  10:00에 진료 시작  방문자 리뷰 321   
19  김성희마리안산부인과의원     서울 송파구 신천동     예약  09:30에 진료 시작  방문자 리뷰 284   

           블로그리뷰    산부인과전문의수  기타전문의여부  
0       블로그 리뷰 3  산부인과전문의 1명      NaN  
1     블로그 리뷰 650  산부인과전문의 1명      NaN  
2       블로그 리뷰 3  산부인과전문의 1명      NaN  
3     블로그 리뷰 141  산부인과전문의 1명      NaN  
4       블로그 리뷰 2  산부인과전문의 1명      NaN  
5       블로그 리뷰 4  산부인과전문의 1명      NaN  
6       블로그 리뷰 2  산부인과전문의 1명      NaN  
7       블로그 리뷰 4  산부인과전문의 1명      NaN  
8       블로그 리뷰 3  산부인과전문의 1명      NaN  
9       블로그 리뷰 4  산부인과전문의 1명      NaN  
10     블로그 리뷰 13  산부인과전문의 1명      NaN  
11      블로그 리뷰 6  산부인과전문의 1명      NaN  
12      블로그 리뷰 1  산부인과전문의 1명      NaN  
13    블로그 리뷰 432  산부인과전문의 1명      NaN  
14    블로그 리뷰 126  산부인과전문의 1명      NaN  
15     블로그 리뷰 24  산부인과전문의 1명      NaN  
16    블로그 리뷰 664         NaN      NaN  
17  블로그 리뷰 1,154  산부인과전문의 1명      NaN  
18      블로그 리뷰 2  산부인과전문의 1명      NaN  
19      블로그 리뷰 3  산부인과전문의 1명      NaN

# 최종 데이터 merging

In [ ]:
# 두 데이터프레임을 병합합니다 (기본적으로 아래쪽으로 병합됨)
merged_df12345 = pd.concat([merged_df1234, df5], ignore_index=True)
merged_df12345

병원명             주소 예약가능여부        진료시작시각         방문자리뷰  \
0       김은주산부인과의원    서울 구로구 신도림동    NaN           NaN           NaN   
1      나를위한산부인과의원     서울 강남구 역삼동     예약           NaN           NaN   
2    애플산부인과의원 강남점     서울 서초구 서초동     예약  10:00에 진료 시작  방문자 리뷰 1,305   
3       에이린산부인과의원     서울 서초구 서초동     예약  10:00에 진료 시작    방문자 리뷰 394   
4     애플산부인과의원 잠실     서울 송파구 잠실동     예약      07/23 휴무    방문자 리뷰 958   
..            ...            ...    ...           ...           ...   
307      봄꽃산부인과의원   서울 서대문구 북가좌동     예약  09:30에 진료 시작     방문자 리뷰 26   
308    더행복한산부인과의원  서울 성북구 동소문동5가    NaN  10:00에 진료 시작    방문자 리뷰 508   
309     여미애산부인과의원      서울 양천구 목동     예약  09:30에 진료 시작    방문자 리뷰 427   
310        우디여성의원      서울 양천구 목동    NaN  10:00에 진료 시작    방문자 리뷰 321   
311  김성희마리안산부인과의원     서울 송파구 신천동     예약  09:30에 진료 시작    방문자 리뷰 284   

            블로그리뷰    산부인과전문의수 기타전문의여부  
0             NaN         NaN     NaN  
1             NaN         NaN     NaN  
2      블로그 리뷰 650  산부인과전문의 3명     NaN  
3    블로그 리뷰 1,353  산부인과전문의 1명     NaN  
4      블로그 리뷰 824  산부인과전문의 2명     NaN  
..            ...         ...     ...  
307     블로그 리뷰 24  산부인과전문의 1명     NaN  
308    블로그 리뷰 664         NaN     NaN  
309  블로그 리뷰 1,154  산부인과전문의 1명     NaN  
310      블로그 리뷰 2  산부인과전문의 1명     NaN  
311      블로그 리뷰 3  산부인과전문의 1명     NaN  

[312 rows x 8 columns]

In [ ]:
# 파일다운
from google.colab import files

# 변경된 데이터를 엑셀 파일로 저장합니다
merged_df12345.to_excel('merged_df12345.xlsx', index=False)

# 저장한 파일을 다운로드할 수 있도록 합니다
files.download('merged_df12345.xlsx')

<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>

## 최종데이터 및 데이터 보충
- 최종데이터에 NaN값 채울 수 있을만큼 채워둔 merged_df_fixing.xlsx
- merged_df_fixing을 기반으로 json 추출하기
- Ascii, UTF 둘다 할 것

In [ ]:
df_fin = pd.read_excel('./merged_df_fixing.xlsx')
df_fin

병원명             주소 예약가능여부        진료시작시각         방문자리뷰  \
0       김은주산부인과의원    서울 구로구 신도림동    NaN  10:00에 진료 시작    방문자 리뷰 428   
1      나를위한산부인과의원     서울 강남구 역삼동     예약  10:00에 진료 시작    방문자 리뷰 770   
2    애플산부인과의원 강남점     서울 서초구 서초동     예약  10:00에 진료 시작  방문자 리뷰 1,305   
3       에이린산부인과의원     서울 서초구 서초동     예약  10:00에 진료 시작    방문자 리뷰 394   
4     애플산부인과의원 잠실     서울 송파구 잠실동     예약  10:00에 진료 시작    방문자 리뷰 958   
..            ...            ...    ...           ...           ...   
307      봄꽃산부인과의원   서울 서대문구 북가좌동     예약  09:30에 진료 시작     방문자 리뷰 26   
308    더행복한산부인과의원  서울 성북구 동소문동5가    NaN  10:00에 진료 시작    방문자 리뷰 508   
309     여미애산부인과의원      서울 양천구 목동     예약  09:30에 진료 시작    방문자 리뷰 427   
310        우디여성의원      서울 양천구 목동    NaN  10:00에 진료 시작    방문자 리뷰 321   
311  김성희마리안산부인과의원     서울 송파구 신천동     예약  09:30에 진료 시작    방문자 리뷰 284   

            블로그리뷰    산부인과전문의수 기타전문의여부  
0      블로그 리뷰 116  산부인과전문의 1명     NaN  
1    블로그 리뷰 1,606  산부인과전문의 6명     NaN  
2      블로그 리뷰 650  산부인과전문의 3명     NaN  
3    블로그 리뷰 1,353  산부인과전문의 1명     NaN  
4      블로그 리뷰 824  산부인과전문의 2명     NaN  
..            ...         ...     ...  
307     블로그 리뷰 24  산부인과전문의 1명     NaN  
308    블로그 리뷰 664         NaN     NaN  
309  블로그 리뷰 1,154  산부인과전문의 1명     NaN  
310      블로그 리뷰 2  산부인과전문의 1명     NaN  
311      블로그 리뷰 3  산부인과전문의 1명     NaN  

[312 rows x 8 columns]

In [ ]:
# JSON 파일로 저장합니다
df_fin.to_json('merged_df_Ascii.json', orient='records', lines=True)
print("JSON으로 변환 완료_Ascii")
files.download('merged_df_Ascii.json')

JSON으로 변환 완료_Ascii


<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>

In [ ]:
# json 변환 시 UTF-8로 줘야함
# 데이터프레임을 JSON 파일로 저장 (UTF-8 인코딩)
df_fin.to_json('merged_df_UTF.json', orient='records', lines=True, force_ascii=False)

# 저장한 파일을 Colab에서 다운로드할 수 있도록 합니다 (필요 시)
from google.colab import files
files.download('merged_df_UTF.json')

<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>

In [ ]:
#그냥 xlsx 받아두기
# 변경된 데이터를 엑셀 파일로 저장합니다
df_fin.to_excel('merged_df.xlsx', index=False)

# 저장한 파일을 다운로드할 수 있도록 합니다
files.download('merged_df.xlsx')

<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>

# json 양식 수정

In [ ]:
import pandas as pd

# XLSX 파일을 읽습니다
df = pd.read_excel('merged_df.xlsx')

# 데이터프레임을 JSON 형식으로 변환 (각 행을 JSON 객체로 변환)
json_data = df.to_json(orient='records', force_ascii=False)

# JSON 데이터를 파일로 저장
with open('hospital_list.json', 'w', encoding='utf-8') as f:
    f.write(json_data)

# 코랩에서 파일을 다운로드 링크로 제공
from google.colab import files
files.download('hospital_list.json')

<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>

## 주소마다 구만 추출하기

In [ ]:
import pandas as pd
import json

df = pd.read_json('hospital_list.json')
df

병원명             주소 예약가능여부        진료시작시각         방문자리뷰  \
0       김은주산부인과의원    서울 구로구 신도림동   None  10:00에 진료 시작    방문자 리뷰 428   
1      나를위한산부인과의원     서울 강남구 역삼동     예약  10:00에 진료 시작    방문자 리뷰 770   
2    애플산부인과의원 강남점     서울 서초구 서초동     예약  10:00에 진료 시작  방문자 리뷰 1,305   
3       에이린산부인과의원     서울 서초구 서초동     예약  10:00에 진료 시작    방문자 리뷰 394   
4     애플산부인과의원 잠실     서울 송파구 잠실동     예약  10:00에 진료 시작    방문자 리뷰 958   
..            ...            ...    ...           ...           ...   
307      봄꽃산부인과의원   서울 서대문구 북가좌동     예약  09:30에 진료 시작     방문자 리뷰 26   
308    더행복한산부인과의원  서울 성북구 동소문동5가   None  10:00에 진료 시작    방문자 리뷰 508   
309     여미애산부인과의원      서울 양천구 목동     예약  09:30에 진료 시작    방문자 리뷰 427   
310        우디여성의원      서울 양천구 목동   None  10:00에 진료 시작    방문자 리뷰 321   
311  김성희마리안산부인과의원     서울 송파구 신천동     예약  09:30에 진료 시작    방문자 리뷰 284   

            블로그리뷰    산부인과전문의수 기타전문의여부  
0      블로그 리뷰 116  산부인과전문의 1명    None  
1    블로그 리뷰 1,606  산부인과전문의 6명    None  
2      블로그 리뷰 650  산부인과전문의 3명    None  
3    블로그 리뷰 1,353  산부인과전문의 1명    None  
4      블로그 리뷰 824  산부인과전문의 2명    None  
..            ...         ...     ...  
307     블로그 리뷰 24  산부인과전문의 1명    None  
308    블로그 리뷰 664        None    None  
309  블로그 리뷰 1,154  산부인과전문의 1명    None  
310      블로그 리뷰 2  산부인과전문의 1명    None  
311      블로그 리뷰 3  산부인과전문의 1명    None  

[312 rows x 8 columns]

In [ ]:
# 주소에서 구 정보를 추출하는 함수
def extract_gu(address):
    if isinstance(address, str):
        split_address = address.split()
        for part in split_address:
            if part.endswith('구'):
                return part
    return None

# 구 정보를 추출하여 새로운 열로 추가
df['구'] = df['주소'].apply(extract_gu)
df

병원명             주소 예약가능여부        진료시작시각         방문자리뷰  \
0       김은주산부인과의원    서울 구로구 신도림동   None  10:00에 진료 시작    방문자 리뷰 428   
1      나를위한산부인과의원     서울 강남구 역삼동     예약  10:00에 진료 시작    방문자 리뷰 770   
2    애플산부인과의원 강남점     서울 서초구 서초동     예약  10:00에 진료 시작  방문자 리뷰 1,305   
3       에이린산부인과의원     서울 서초구 서초동     예약  10:00에 진료 시작    방문자 리뷰 394   
4     애플산부인과의원 잠실     서울 송파구 잠실동     예약  10:00에 진료 시작    방문자 리뷰 958   
..            ...            ...    ...           ...           ...   
307      봄꽃산부인과의원   서울 서대문구 북가좌동     예약  09:30에 진료 시작     방문자 리뷰 26   
308    더행복한산부인과의원  서울 성북구 동소문동5가   None  10:00에 진료 시작    방문자 리뷰 508   
309     여미애산부인과의원      서울 양천구 목동     예약  09:30에 진료 시작    방문자 리뷰 427   
310        우디여성의원      서울 양천구 목동   None  10:00에 진료 시작    방문자 리뷰 321   
311  김성희마리안산부인과의원     서울 송파구 신천동     예약  09:30에 진료 시작    방문자 리뷰 284   

            블로그리뷰    산부인과전문의수 기타전문의여부     구  
0      블로그 리뷰 116  산부인과전문의 1명    None   구로구  
1    블로그 리뷰 1,606  산부인과전문의 6명    None   강남구  
2      블로그 리뷰 650  산부인과전문의 3명    None   서초구  
3    블로그 리뷰 1,353  산부인과전문의 1명    None   서초구  
4      블로그 리뷰 824  산부인과전문의 2명    None   송파구  
..            ...         ...     ...   ...  
307     블로그 리뷰 24  산부인과전문의 1명    None  서대문구  
308    블로그 리뷰 664        None    None   성북구  
309  블로그 리뷰 1,154  산부인과전문의 1명    None   양천구  
310      블로그 리뷰 2  산부인과전문의 1명    None   양천구  
311      블로그 리뷰 3  산부인과전문의 1명    None   송파구  

[312 rows x 9 columns]

In [ ]:
#컬럼순서 맞춰주기
cols = ['병원명', '주소', '구','예약가능여부', '진료시작시각', '방문자리뷰', '블로그리뷰', '산부인과전문의수', '기타전문의여부']
df = df[cols]
df

병원명             주소     구 예약가능여부        진료시작시각         방문자리뷰  \
0       김은주산부인과의원    서울 구로구 신도림동   구로구   None  10:00에 진료 시작    방문자 리뷰 428   
1      나를위한산부인과의원     서울 강남구 역삼동   강남구     예약  10:00에 진료 시작    방문자 리뷰 770   
2    애플산부인과의원 강남점     서울 서초구 서초동   서초구     예약  10:00에 진료 시작  방문자 리뷰 1,305   
3       에이린산부인과의원     서울 서초구 서초동   서초구     예약  10:00에 진료 시작    방문자 리뷰 394   
4     애플산부인과의원 잠실     서울 송파구 잠실동   송파구     예약  10:00에 진료 시작    방문자 리뷰 958   
..            ...            ...   ...    ...           ...           ...   
307      봄꽃산부인과의원   서울 서대문구 북가좌동  서대문구     예약  09:30에 진료 시작     방문자 리뷰 26   
308    더행복한산부인과의원  서울 성북구 동소문동5가   성북구   None  10:00에 진료 시작    방문자 리뷰 508   
309     여미애산부인과의원      서울 양천구 목동   양천구     예약  09:30에 진료 시작    방문자 리뷰 427   
310        우디여성의원      서울 양천구 목동   양천구   None  10:00에 진료 시작    방문자 리뷰 321   
311  김성희마리안산부인과의원     서울 송파구 신천동   송파구     예약  09:30에 진료 시작    방문자 리뷰 284   

            블로그리뷰    산부인과전문의수 기타전문의여부  
0      블로그 리뷰 116  산부인과전문의 1명    None  
1    블로그 리뷰 1,606  산부인과전문의 6명    None  
2      블로그 리뷰 650  산부인과전문의 3명    None  
3    블로그 리뷰 1,353  산부인과전문의 1명    None  
4      블로그 리뷰 824  산부인과전문의 2명    None  
..            ...         ...     ...  
307     블로그 리뷰 24  산부인과전문의 1명    None  
308    블로그 리뷰 664        None    None  
309  블로그 리뷰 1,154  산부인과전문의 1명    None  
310      블로그 리뷰 2  산부인과전문의 1명    None  
311      블로그 리뷰 3  산부인과전문의 1명    None  

[312 rows x 9 columns]

In [ ]:
# 데이터프레임을 JSON 형식으로 변환 (각 행을 JSON 객체로 변환)
json_data = df.to_json(orient='records', force_ascii=False)

# JSON 데이터를 파일로 저장
with open('hospital_list.json', 'w', encoding='utf-8') as f:
    f.write(json_data)

# 코랩에서 파일을 다운로드 링크로 제공
from google.colab import files
files.download('hospital_list.json')

<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>

# json 구조 수정
- 0728 요청
- {
    "id": 1,
    "hospital_name": "김은주산부인과의원",
    "address": "서울 구로구 신도림동",
    "gu": "구로구",
    "reservation": null,
    "visitcnt": "428",
    "blogcnt": "116",
    "maindoctorcnt": "1"
}

In [9]:
import pandas as pd
import json

df_fix = pd.read_json('hospital_list_withGU.json')
df_fix

병원명             주소     구 예약가능여부        진료시작시각         방문자리뷰  \
0       김은주산부인과의원    서울 구로구 신도림동   구로구   None  10:00에 진료 시작    방문자 리뷰 428   
1      나를위한산부인과의원     서울 강남구 역삼동   강남구     예약  10:00에 진료 시작    방문자 리뷰 770   
2    애플산부인과의원 강남점     서울 서초구 서초동   서초구     예약  10:00에 진료 시작  방문자 리뷰 1,305   
3       에이린산부인과의원     서울 서초구 서초동   서초구     예약  10:00에 진료 시작    방문자 리뷰 394   
4     애플산부인과의원 잠실     서울 송파구 잠실동   송파구     예약  10:00에 진료 시작    방문자 리뷰 958   
..            ...            ...   ...    ...           ...           ...   
307      봄꽃산부인과의원   서울 서대문구 북가좌동  서대문구     예약  09:30에 진료 시작     방문자 리뷰 26   
308    더행복한산부인과의원  서울 성북구 동소문동5가   성북구   None  10:00에 진료 시작    방문자 리뷰 508   
309     여미애산부인과의원      서울 양천구 목동   양천구     예약  09:30에 진료 시작    방문자 리뷰 427   
310        우디여성의원      서울 양천구 목동   양천구   None  10:00에 진료 시작    방문자 리뷰 321   
311  김성희마리안산부인과의원     서울 송파구 신천동   송파구     예약  09:30에 진료 시작    방문자 리뷰 284   

            블로그리뷰    산부인과전문의수 기타전문의여부  
0      블로그 리뷰 116  산부인과전문의 1명    None  
1    블로그 리뷰 1,606  산부인과전문의 6명    None  
2      블로그 리뷰 650  산부인과전문의 3명    None  
3    블로그 리뷰 1,353  산부인과전문의 1명    None  
4      블로그 리뷰 824  산부인과전문의 2명    None  
..            ...         ...     ...  
307     블로그 리뷰 24  산부인과전문의 1명    None  
308    블로그 리뷰 664        None    None  
309  블로그 리뷰 1,154  산부인과전문의 1명    None  
310      블로그 리뷰 2  산부인과전문의 1명    None  
311      블로그 리뷰 3  산부인과전문의 1명    None  

[312 rows x 9 columns]

In [11]:
df_ren = df_fix.rename(columns = {
    "병원명":"hospital_name",
    "주소":"address",
    "구":"gu",
    "예약가능여부":"reservation",
    "방문자리뷰":"visitcnt",
    "블로그리뷰":"blogcnt",
    "산부인과전문의수":"maindoctorcnt"})

In [12]:
df_ren

hospital_name        address    gu reservation        진료시작시각  \
0       김은주산부인과의원    서울 구로구 신도림동   구로구        None  10:00에 진료 시작   
1      나를위한산부인과의원     서울 강남구 역삼동   강남구          예약  10:00에 진료 시작   
2    애플산부인과의원 강남점     서울 서초구 서초동   서초구          예약  10:00에 진료 시작   
3       에이린산부인과의원     서울 서초구 서초동   서초구          예약  10:00에 진료 시작   
4     애플산부인과의원 잠실     서울 송파구 잠실동   송파구          예약  10:00에 진료 시작   
..            ...            ...   ...         ...           ...   
307      봄꽃산부인과의원   서울 서대문구 북가좌동  서대문구          예약  09:30에 진료 시작   
308    더행복한산부인과의원  서울 성북구 동소문동5가   성북구        None  10:00에 진료 시작   
309     여미애산부인과의원      서울 양천구 목동   양천구          예약  09:30에 진료 시작   
310        우디여성의원      서울 양천구 목동   양천구        None  10:00에 진료 시작   
311  김성희마리안산부인과의원     서울 송파구 신천동   송파구          예약  09:30에 진료 시작   

         visitcnt       blogcnt maindoctorcnt 기타전문의여부  
0      방문자 리뷰 428    블로그 리뷰 116    산부인과전문의 1명    None  
1      방문자 리뷰 770  블로그 리뷰 1,606    산부인과전문의 6명    None  
2    방문자 리뷰 1,305    블로그 리뷰 650    산부인과전문의 3명    None  
3      방문자 리뷰 394  블로그 리뷰 1,353    산부인과전문의 1명    None  
4      방문자 리뷰 958    블로그 리뷰 824    산부인과전문의 2명    None  
..            ...           ...           ...     ...  
307     방문자 리뷰 26     블로그 리뷰 24    산부인과전문의 1명    None  
308    방문자 리뷰 508    블로그 리뷰 664          None    None  
309    방문자 리뷰 427  블로그 리뷰 1,154    산부인과전문의 1명    None  
310    방문자 리뷰 321      블로그 리뷰 2    산부인과전문의 1명    None  
311    방문자 리뷰 284      블로그 리뷰 3    산부인과전문의 1명    None  

[312 rows x 9 columns]

In [15]:
# 진료시작시각, 기타전문의 여부 컬럼 drop
df_ren = df_ren.drop(columns = ['진료시작시각','기타전문의여부'])
df_ren

hospital_name        address    gu reservation      visitcnt  \
0       김은주산부인과의원    서울 구로구 신도림동   구로구        None    방문자 리뷰 428   
1      나를위한산부인과의원     서울 강남구 역삼동   강남구          예약    방문자 리뷰 770   
2    애플산부인과의원 강남점     서울 서초구 서초동   서초구          예약  방문자 리뷰 1,305   
3       에이린산부인과의원     서울 서초구 서초동   서초구          예약    방문자 리뷰 394   
4     애플산부인과의원 잠실     서울 송파구 잠실동   송파구          예약    방문자 리뷰 958   
..            ...            ...   ...         ...           ...   
307      봄꽃산부인과의원   서울 서대문구 북가좌동  서대문구          예약     방문자 리뷰 26   
308    더행복한산부인과의원  서울 성북구 동소문동5가   성북구        None    방문자 리뷰 508   
309     여미애산부인과의원      서울 양천구 목동   양천구          예약    방문자 리뷰 427   
310        우디여성의원      서울 양천구 목동   양천구        None    방문자 리뷰 321   
311  김성희마리안산부인과의원     서울 송파구 신천동   송파구          예약    방문자 리뷰 284   

          blogcnt maindoctorcnt  
0      블로그 리뷰 116    산부인과전문의 1명  
1    블로그 리뷰 1,606    산부인과전문의 6명  
2      블로그 리뷰 650    산부인과전문의 3명  
3    블로그 리뷰 1,353    산부인과전문의 1명  
4      블로그 리뷰 824    산부인과전문의 2명  
..            ...           ...  
307     블로그 리뷰 24    산부인과전문의 1명  
308    블로그 리뷰 664          None  
309  블로그 리뷰 1,154    산부인과전문의 1명  
310      블로그 리뷰 2    산부인과전문의 1명  
311      블로그 리뷰 3    산부인과전문의 1명  

[312 rows x 7 columns]

In [17]:
# '방문자리뷰'와 '블로그리뷰' 컬럼에서 숫자만 추출
df_ren['visitcnt'] = df_ren['visitcnt'].str.extract('(\d+)').fillna(0).astype(int)
df_ren['blogcnt'] = df_ren['blogcnt'].str.extract('(\d+)').fillna(0).astype(int)
df_ren['maindoctorcnt'] = df_ren['maindoctorcnt'].str.extract('(\d+)').fillna(0).astype(int)
df_ren

hospital_name        address    gu reservation  visitcnt  blogcnt  \
0       김은주산부인과의원    서울 구로구 신도림동   구로구        None       428      116   
1      나를위한산부인과의원     서울 강남구 역삼동   강남구          예약       770        1   
2    애플산부인과의원 강남점     서울 서초구 서초동   서초구          예약         1      650   
3       에이린산부인과의원     서울 서초구 서초동   서초구          예약       394        1   
4     애플산부인과의원 잠실     서울 송파구 잠실동   송파구          예약       958      824   
..            ...            ...   ...         ...       ...      ...   
307      봄꽃산부인과의원   서울 서대문구 북가좌동  서대문구          예약        26       24   
308    더행복한산부인과의원  서울 성북구 동소문동5가   성북구        None       508      664   
309     여미애산부인과의원      서울 양천구 목동   양천구          예약       427        1   
310        우디여성의원      서울 양천구 목동   양천구        None       321        2   
311  김성희마리안산부인과의원     서울 송파구 신천동   송파구          예약       284        3   

     maindoctorcnt  
0                1  
1                6  
2                3  
3                1  
4                2  
..             ...  
307              1  
308              0  
309              1  
310              1  
311              1  

[312 rows x 7 columns]

In [19]:
# 맨앞 컬럼에 id 추가
df_ren['id'] = range(1,len(df_ren) + 1)
df_ren

hospital_name        address    gu reservation  visitcnt  blogcnt  \
0       김은주산부인과의원    서울 구로구 신도림동   구로구        None       428      116   
1      나를위한산부인과의원     서울 강남구 역삼동   강남구          예약       770        1   
2    애플산부인과의원 강남점     서울 서초구 서초동   서초구          예약         1      650   
3       에이린산부인과의원     서울 서초구 서초동   서초구          예약       394        1   
4     애플산부인과의원 잠실     서울 송파구 잠실동   송파구          예약       958      824   
..            ...            ...   ...         ...       ...      ...   
307      봄꽃산부인과의원   서울 서대문구 북가좌동  서대문구          예약        26       24   
308    더행복한산부인과의원  서울 성북구 동소문동5가   성북구        None       508      664   
309     여미애산부인과의원      서울 양천구 목동   양천구          예약       427        1   
310        우디여성의원      서울 양천구 목동   양천구        None       321        2   
311  김성희마리안산부인과의원     서울 송파구 신천동   송파구          예약       284        3   

     maindoctorcnt   id  
0                1    1  
1                6    2  
2                3    3  
3                1    4  
4                2    5  
..             ...  ...  
307              1  308  
308              0  309  
309              1  310  
310              1  311  
311              1  312  

[312 rows x 8 columns]

In [20]:
# 컬럼순서배치
df = df_ren[['id', 'hospital_name', 'address', 'gu', 'reservation', 'visitcnt', 'blogcnt', 'maindoctorcnt']]
df

id hospital_name        address    gu reservation  visitcnt  blogcnt  \
0      1     김은주산부인과의원    서울 구로구 신도림동   구로구        None       428      116   
1      2    나를위한산부인과의원     서울 강남구 역삼동   강남구          예약       770        1   
2      3  애플산부인과의원 강남점     서울 서초구 서초동   서초구          예약         1      650   
3      4     에이린산부인과의원     서울 서초구 서초동   서초구          예약       394        1   
4      5   애플산부인과의원 잠실     서울 송파구 잠실동   송파구          예약       958      824   
..   ...           ...            ...   ...         ...       ...      ...   
307  308      봄꽃산부인과의원   서울 서대문구 북가좌동  서대문구          예약        26       24   
308  309    더행복한산부인과의원  서울 성북구 동소문동5가   성북구        None       508      664   
309  310     여미애산부인과의원      서울 양천구 목동   양천구          예약       427        1   
310  311        우디여성의원      서울 양천구 목동   양천구        None       321        2   
311  312  김성희마리안산부인과의원     서울 송파구 신천동   송파구          예약       284        3   

     maindoctorcnt  
0                1  
1                6  
2                3  
3                1  
4                2  
..             ...  
307              1  
308              0  
309              1  
310              1  
311              1  

[312 rows x 8 columns]

In [21]:
# 데이터프레임을 JSON 형식으로 변환 (각 행을 JSON 객체로 변환)
json_data = df.to_json(orient='records', force_ascii=False)

# JSON 데이터를 파일로 저장
with open('hospital_fix.json', 'w', encoding='utf-8') as f:
    f.write(json_data)

# 코랩에서 파일을 다운로드 링크로 제공
from google.colab import files
files.download('hospital_fix.json')

<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>

# 새작업